# Analysis

## Preliminary

``` r

## load R package
library(msDiaLogue)
## preprocessing
fileName <- "../inst/extdata/Toy_Spectronaut_Data.csv"
dataSet <- preprocessing(fileName,
                         filterNaN = TRUE, filterUnique = 2,
                         replaceBlank = TRUE, saveRm = TRUE)
## transformation
dataTran <- transform(dataSet, logFold = 2)
## normalization
dataNorm <- normalize(dataTran, normalizeType = "quant")
## imputation
dataImput <- impute.min_local(dataNorm, reportImputing = FALSE,
                              reqPercentPresent = 0.51)
## filtering
dataImput <- filterNA(dataImput, saveRm = TRUE)
```

The functions in the analysis module calculate the results that can be
used in subsequent visualizations.

**Note:** The following analyses compare all other conditions against
the reference condition, which is specified by the argument `ref`, for
multiple comparisons. If `ref` is not provided, it will be automatically
generated by all the combinations of two conditions, based on the level
attributes of the condition.  
For example, suppose there are three conditions in the data: “A”, “B”,
and “C”. If you specify `ref = "A"`, then the result includes two
comparisons: “B-A” and “C-A”. If `ref = NULL`, there will be three
comparisons: “A-B”, “A-C”, and “B-C”.

## Student’s t-test

### Example

``` r

anlys_t <- analyze.t(dataImput, ref = "50pmol", adjust.method = "none",
                     saveRes = TRUE)
#> Data are essentially constant.
#> Data are essentially constant.
```

**Note:** In the Student’s t-test, a warning message might appear,
stating “**Data are essentially constant**,” which means that the data
contain proteins with the same value in all samples. In this case, the
p-value of t-test returns NaN.

    #> $`100pmol-50pmol`

|  | NUD4B_HUMAN (+1) | A0A7P0T808_HUMAN (+1) | A0A8I5KU53_HUMAN (+1) | ZN840_HUMAN | CC85C_HUMAN | C9JEV0_HUMAN (+1) | C9JNU9_HUMAN | ALBU_BOVIN | CYC_BOVIN | TRFE_BOVIN | F8W0H2_HUMAN | H0Y7V7_HUMAN (+1) | H0YD14_HUMAN | H3BUF6_HUMAN | H7C1W4_HUMAN (+1) | H7C3M7_HUMAN | TLR3_HUMAN | LRIG2_HUMAN | RAB3D_HUMAN | ADH1_YEAST | LYSC_CHICK | BGAL_ECOLI | CYTA_HUMAN | KPCB_HUMAN | LIPL_HUMAN | CO6_HUMAN | BGAL_HUMAN | SYTC_HUMAN | CASPE_HUMAN | DCAF6_HUMAN | DALD3_HUMAN | HGNAT_HUMAN | RFFL_HUMAN | RN185_HUMAN | ZN462_HUMAN | ALKB7_HUMAN | POLK_HUMAN | ACAD8_HUMAN |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| 100pmol mean | 10.6532858 | 12.0862074 | 10.6232407 | 8.2644365 | 8.8486965 | 6.8764321 | 8.1789684 | 16.75777 | 13.0941802 | 14.0233390 | 9.0659703 | 10.0672927 | 9.9408634 | 7.9781022 | 7.8297313 | 9.8230915 | 7.4096023 | 8.5986327 | 10.0905795 | 15.1604531 | 13.6952175 | 14.3757288 | 8.892464 | 9.8735187 | 8.2974346 | 6.7681823 | 11.7142153 | 14.7560796 | 6.3848929 | 7.9228836 | 9.6214990 | 10.6043210 | 7.0043419 | 10.0418355 | 8.6778910 | 6.8092287 | 7.8188307 | 7.5025180 |
| 50pmol mean | 11.0652499 | 7.8125342 | 9.9553033 | 7.9185075 | 8.6105032 | 8.4512816 | 8.3913880 | 16.75777 | 12.6233713 | 13.9262822 | 9.6052951 | 9.6068208 | 10.3568778 | 8.5314736 | 6.9189678 | 9.8860972 | 7.1716461 | 8.6447744 | 10.4687736 | 14.6152277 | 13.4461368 | 14.2433512 | 10.278260 | 10.2093348 | 8.6777501 | 7.2864883 | 11.7603897 | 15.1244369 | 8.3934284 | 7.6429483 | 8.6746639 | 9.8868633 | 6.7178161 | 9.3454998 | 8.2512331 | 7.9571364 | 6.6190600 | 6.8997407 |
| difference | -0.4119641 | 4.2736732 | 0.6679373 | 0.3459290 | 0.2381933 | -1.5748494 | -0.2124196 | 0.00000 | 0.4708090 | 0.0970569 | -0.5393248 | 0.4604719 | -0.4160144 | -0.5533715 | 0.9107635 | -0.0630057 | 0.2379563 | -0.0461417 | -0.3781941 | 0.5452255 | 0.2490808 | 0.1323776 | -1.385796 | -0.3358160 | -0.3803155 | -0.5183059 | -0.0461744 | -0.3683573 | -2.0085355 | 0.2799353 | 0.9468351 | 0.7174576 | 0.2865258 | 0.6963357 | 0.4266578 | -1.1479077 | 1.1997707 | 0.6027773 |
| p-value | 0.2425281 | 0.0223332 | 0.4450987 | 0.5455559 | 0.1909446 | 0.0348158 | 0.3685157 | NaN | 0.1922051 | 0.3275609 | 0.0005957 | 0.0767642 | 0.0683551 | 0.0041165 | 0.0651420 | 0.8119403 | 0.2805476 | 0.8086313 | 0.0024306 | 0.0076435 | 0.2145975 | 0.2343297 | 0.009692 | 0.0658082 | 0.1068935 | 0.0481610 | 0.8918516 | 0.0000000 | 0.0232926 | 0.3007562 | 0.0633640 | 0.0942018 | 0.1959323 | 0.0503672 | 0.5680738 | 0.0436514 | 0.1120976 | 0.0022576 |

    #> $`200pmol-50pmol`

|  | NUD4B_HUMAN (+1) | A0A7P0T808_HUMAN (+1) | A0A8I5KU53_HUMAN (+1) | ZN840_HUMAN | CC85C_HUMAN | C9JEV0_HUMAN (+1) | C9JNU9_HUMAN | ALBU_BOVIN | CYC_BOVIN | TRFE_BOVIN | F8W0H2_HUMAN | H0Y7V7_HUMAN (+1) | H0YD14_HUMAN | H3BUF6_HUMAN | H7C1W4_HUMAN (+1) | H7C3M7_HUMAN | TLR3_HUMAN | LRIG2_HUMAN | RAB3D_HUMAN | ADH1_YEAST | LYSC_CHICK | BGAL_ECOLI | CYTA_HUMAN | KPCB_HUMAN | LIPL_HUMAN | CO6_HUMAN | BGAL_HUMAN | SYTC_HUMAN | CASPE_HUMAN | DCAF6_HUMAN | DALD3_HUMAN | HGNAT_HUMAN | RFFL_HUMAN | RN185_HUMAN | ZN462_HUMAN | ALKB7_HUMAN | POLK_HUMAN | ACAD8_HUMAN |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| 200pmol mean | 10.6488941 | 9.9995022 | 11.2485095 | 8.4505628 | 8.2445037 | 6.7465842 | 8.3091726 | 16.75777 | 13.2471248 | 13.7426617 | 9.2570565 | 10.1413903 | 9.9978891 | 8.0880583 | 7.5631150 | 10.0415877 | 7.5210813 | 8.4008606 | 10.2414149 | 15.1516556 | 14.1141044 | 14.7415973 | 8.8133014 | 10.1732192 | 8.4520910 | 7.0971425 | 12.1775775 | 14.4198014 | 6.1065048 | 7.4951113 | 9.9165721 | 10.5958169 | 6.7744045 | 9.4079382 | 8.8671576 | 7.5851679 | 7.2449244 | 8.1196440 |
| 50pmol mean | 11.0652499 | 7.8125342 | 9.9553033 | 7.9185075 | 8.6105032 | 8.4512816 | 8.3913880 | 16.75777 | 12.6233713 | 13.9262822 | 9.6052951 | 9.6068208 | 10.3568778 | 8.5314736 | 6.9189678 | 9.8860972 | 7.1716461 | 8.6447744 | 10.4687736 | 14.6152277 | 13.4461368 | 14.2433512 | 10.2782600 | 10.2093348 | 8.6777501 | 7.2864883 | 11.7603897 | 15.1244369 | 8.3934284 | 7.6429483 | 8.6746639 | 9.8868633 | 6.7178161 | 9.3454998 | 8.2512331 | 7.9571364 | 6.6190600 | 6.8997407 |
| difference | -0.4163558 | 2.1869680 | 1.2932062 | 0.5320553 | -0.3659995 | -1.7046973 | -0.0822154 | 0.00000 | 0.6237535 | -0.1836205 | -0.3482386 | 0.5345694 | -0.3589887 | -0.4434153 | 0.6441472 | 0.1554905 | 0.3494352 | -0.2439139 | -0.2273586 | 0.5364280 | 0.6679676 | 0.4982461 | -1.4649585 | -0.0361156 | -0.2256591 | -0.1893458 | 0.4171878 | -0.7046356 | -2.2869237 | -0.1478371 | 1.2419082 | 0.7089535 | 0.0565885 | 0.0624383 | 0.6159244 | -0.3719685 | 0.6258644 | 1.2199033 |
| p-value | 0.1446884 | 0.1826838 | 0.1875421 | 0.3662078 | 0.3927857 | 0.0181442 | 0.6911235 | NaN | 0.1771520 | 0.2818360 | 0.0319217 | 0.0503531 | 0.0448461 | 0.0166531 | 0.1885601 | 0.0812056 | 0.0303572 | 0.2404181 | 0.1646796 | 0.0073303 | 0.0124946 | 0.0064381 | 0.0058656 | 0.7824826 | 0.3171743 | 0.3158950 | 0.3983925 | 0.0000458 | 0.0205236 | 0.5114326 | 0.0210858 | 0.2748735 | 0.8040085 | 0.8612941 | 0.4109857 | 0.4319860 | 0.1765836 | 0.0465999 |

### Details

The Student’s t-test is used to compare the means between two conditions
for each protein, reporting both the difference in means between the
conditions and the p-value of the test.

The argument `adjust.method` is used to specify the testing correction
procedure to be applied to p-values. This adjustment is very common in
DNA or RNA-Seq analyses, where datasets are very large and where
researchers are most interested in controlling the Type I error rate
when conducting multiple comparisons.

However, for mass spectrometry-based proteomics results, the dataset
sizes are smaller than in sequencing analyses, and testing corrections
can be too harsh of a threshold to apply. Most often, applying any
testing correction to proteomics data results in there being zero
significant changes. This does not mean that nothing is meaningfully
changing in your dataset. It does mean that these corrections are
usually not a useful tool for finding biologically-relevant changes in
your dataset.

Also keep in mind that reducing Type I error typically comes at the cost
of increasing Type II error, and vice versa. There is no way to
eliminate all error; each researcher must decide whether they are more
comfortable with having more false positives or more false negatives in
the dataset, and choose their analysis strategies accordingly.

UConn PMF recommends not applying testing corrections to your proteomics
dataset, but if you would like to explore the effects of doing so,
several methods are provided below:

1.  “BH” or its alias “fdr”: Benjamini and Hochberg (1995).

2.  “BY”: Benjamini and Yekutieli (2001).

3.  “bonferroni”: Bonferroni (1936).

4.  “hochberg”: Hochberg (1988).

5.  “holm”: Holm (1979).

6.  “hommel”: Hommel (1988).

Each method offers its own balance between statistical power and error
control. The default value `"none"` indicates that no correction is
applied.

## Empirical Bayes moderated t-test

### Example

``` r

anlys_modt <- analyze.mod_t(dataImput, ref = "50pmol", adjust.method = "none",
                            saveRes = TRUE)
#> Warning: Zero sample variances detected, have been offset away from zero
```

**Note:** In the moderated t-test, a warning message might occur
stating, “**Zero sample variances detected, have been offset away from
zero.**” This warning corresponds to examples of proteins that exhibited
identical quant values, either pre- or post-imputation, and therefore no
variance is present across conditions for those proteins. This does not
impede downstream analysis; it merely serves to alert users to its
occurrence.

    #> $`100pmol-50pmol`

|  | NUD4B_HUMAN (+1) | A0A7P0T808_HUMAN (+1) | A0A8I5KU53_HUMAN (+1) | ZN840_HUMAN | CC85C_HUMAN | C9JEV0_HUMAN (+1) | C9JNU9_HUMAN | ALBU_BOVIN | CYC_BOVIN | TRFE_BOVIN | F8W0H2_HUMAN | H0Y7V7_HUMAN (+1) | H0YD14_HUMAN | H3BUF6_HUMAN | H7C1W4_HUMAN (+1) | H7C3M7_HUMAN | TLR3_HUMAN | LRIG2_HUMAN | RAB3D_HUMAN | ADH1_YEAST | LYSC_CHICK | BGAL_ECOLI | CYTA_HUMAN | KPCB_HUMAN | LIPL_HUMAN | CO6_HUMAN | BGAL_HUMAN | SYTC_HUMAN | CASPE_HUMAN | DCAF6_HUMAN | DALD3_HUMAN | HGNAT_HUMAN | RFFL_HUMAN | RN185_HUMAN | ZN462_HUMAN | ALKB7_HUMAN | POLK_HUMAN | ACAD8_HUMAN |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| 100pmol mean | 10.6532858 | 12.0862074 | 10.6232407 | 8.2644365 | 8.8486965 | 6.8764321 | 8.1789684 | 16.75777 | 13.094180 | 14.0233390 | 9.0659703 | 10.0672927 | 9.9408634 | 7.9781022 | 7.8297313 | 9.8230915 | 7.4096023 | 8.5986327 | 10.0905795 | 15.1604531 | 13.6952175 | 14.3757288 | 8.8924636 | 9.8735187 | 8.2974346 | 6.7681823 | 11.7142153 | 14.7560796 | 6.3848929 | 7.9228836 | 9.6214990 | 10.6043210 | 7.0043419 | 10.0418355 | 8.6778910 | 6.8092287 | 7.818831 | 7.5025180 |
| 50pmol mean | 11.0652499 | 7.8125342 | 9.9553033 | 7.9185075 | 8.6105032 | 8.4512816 | 8.3913880 | 16.75777 | 12.623371 | 13.9262822 | 9.6052951 | 9.6068208 | 10.3568778 | 8.5314736 | 6.9189678 | 9.8860972 | 7.1716461 | 8.6447744 | 10.4687736 | 14.6152277 | 13.4461368 | 14.2433512 | 10.2782600 | 10.2093348 | 8.6777501 | 7.2864883 | 11.7603897 | 15.1244369 | 8.3934284 | 7.6429483 | 8.6746639 | 9.8868633 | 6.7178161 | 9.3454998 | 8.2512331 | 7.9571364 | 6.619060 | 6.8997407 |
| difference | -0.4119641 | 4.2736732 | 0.6679373 | 0.3459290 | 0.2381933 | -1.5748494 | -0.2124196 | 0.00000 | 0.470809 | 0.0970569 | -0.5393248 | 0.4604719 | -0.4160144 | -0.5533715 | 0.9107635 | -0.0630057 | 0.2379563 | -0.0461417 | -0.3781941 | 0.5452255 | 0.2490808 | 0.1323776 | -1.3857964 | -0.3358160 | -0.3803155 | -0.5183059 | -0.0461744 | -0.3683573 | -2.0085355 | 0.2799353 | 0.9468351 | 0.7174576 | 0.2865258 | 0.6963357 | 0.4266578 | -1.1479077 | 1.199771 | 0.6027773 |
| p-value | 0.1692657 | 0.0041928 | 0.3594271 | 0.4188776 | 0.4476151 | 0.0077729 | 0.3082409 | 1.00000 | 0.205110 | 0.4719321 | 0.0003809 | 0.0210540 | 0.0273503 | 0.0007011 | 0.0290176 | 0.7547940 | 0.1763304 | 0.7787863 | 0.0090768 | 0.0000260 | 0.1367383 | 0.1511826 | 0.0002994 | 0.0325244 | 0.0523968 | 0.0126619 | 0.9082025 | 0.0000018 | 0.0015443 | 0.2394099 | 0.0184304 | 0.1590979 | 0.1596566 | 0.0381075 | 0.4486484 | 0.0171955 | 0.036571 | 0.0782396 |

    #> $`200pmol-50pmol`

|  | NUD4B_HUMAN (+1) | A0A7P0T808_HUMAN (+1) | A0A8I5KU53_HUMAN (+1) | ZN840_HUMAN | CC85C_HUMAN | C9JEV0_HUMAN (+1) | C9JNU9_HUMAN | ALBU_BOVIN | CYC_BOVIN | TRFE_BOVIN | F8W0H2_HUMAN | H0Y7V7_HUMAN (+1) | H0YD14_HUMAN | H3BUF6_HUMAN | H7C1W4_HUMAN (+1) | H7C3M7_HUMAN | TLR3_HUMAN | LRIG2_HUMAN | RAB3D_HUMAN | ADH1_YEAST | LYSC_CHICK | BGAL_ECOLI | CYTA_HUMAN | KPCB_HUMAN | LIPL_HUMAN | CO6_HUMAN | BGAL_HUMAN | SYTC_HUMAN | CASPE_HUMAN | DCAF6_HUMAN | DALD3_HUMAN | HGNAT_HUMAN | RFFL_HUMAN | RN185_HUMAN | ZN462_HUMAN | ALKB7_HUMAN | POLK_HUMAN | ACAD8_HUMAN |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| 200pmol mean | 10.6488941 | 9.9995022 | 11.2485095 | 8.4505628 | 8.2445037 | 6.7465842 | 8.3091726 | 16.75777 | 13.2471248 | 13.7426617 | 9.2570565 | 10.1413903 | 9.9978891 | 8.0880583 | 7.5631150 | 10.0415877 | 7.5210813 | 8.4008606 | 10.2414149 | 15.1516556 | 14.1141044 | 14.7415973 | 8.813301 | 10.1732192 | 8.4520910 | 7.0971425 | 12.1775775 | 14.4198014 | 6.1065048 | 7.4951113 | 9.9165721 | 10.5958169 | 6.7744045 | 9.4079382 | 8.8671576 | 7.5851679 | 7.2449244 | 8.1196440 |
| 50pmol mean | 11.0652499 | 7.8125342 | 9.9553033 | 7.9185075 | 8.6105032 | 8.4512816 | 8.3913880 | 16.75777 | 12.6233713 | 13.9262822 | 9.6052951 | 9.6068208 | 10.3568778 | 8.5314736 | 6.9189678 | 9.8860972 | 7.1716461 | 8.6447744 | 10.4687736 | 14.6152277 | 13.4461368 | 14.2433512 | 10.278260 | 10.2093348 | 8.6777501 | 7.2864883 | 11.7603897 | 15.1244369 | 8.3934284 | 7.6429483 | 8.6746639 | 9.8868633 | 6.7178161 | 9.3454998 | 8.2512331 | 7.9571364 | 6.6190600 | 6.8997407 |
| difference | -0.4163558 | 2.1869680 | 1.2932062 | 0.5320553 | -0.3659995 | -1.7046973 | -0.0822154 | 0.00000 | 0.6237535 | -0.1836205 | -0.3482386 | 0.5345694 | -0.3589887 | -0.4434153 | 0.6441472 | 0.1554905 | 0.3494352 | -0.2439139 | -0.2273586 | 0.5364280 | 0.6679676 | 0.4982461 | -1.464959 | -0.0361156 | -0.2256591 | -0.1893458 | 0.4171878 | -0.7046356 | -2.2869237 | -0.1478371 | 1.2419082 | 0.7089535 | 0.0565885 | 0.0624383 | 0.6159244 | -0.3719685 | 0.6258644 | 1.2199033 |
| p-value | 0.1651853 | 0.0881485 | 0.0926706 | 0.2239144 | 0.2525044 | 0.0049216 | 0.6864892 | 1.00000 | 0.1028334 | 0.1877393 | 0.0069376 | 0.0099405 | 0.0500060 | 0.0031304 | 0.1016953 | 0.4466320 | 0.0584428 | 0.1582493 | 0.0809141 | 0.0000298 | 0.0014873 | 0.0001655 | 0.000195 | 0.7948930 | 0.2207008 | 0.2936324 | 0.3105040 | 0.0000000 | 0.0006214 | 0.5236720 | 0.0042098 | 0.1636145 | 0.7701943 | 0.8346277 | 0.2815464 | 0.3767021 | 0.2366530 | 0.0026681 |

### Details

The main distinction between the Student’s and empirical Bayes moderated
t-tests (Smyth 2004) lies in how variance is computed. While the
Student’s t-test calculates variance based on the data available for
each protein individually (which will be limited by the number of
replicates included for each condition), the moderated t-test utilizes
information from all replicates of every protein in the current dataset
to calculate variance.

## Wilcoxon test

### Example

``` r

anlys_wilcox <- analyze.wilcox(dataImput, ref = "50pmol", adjust.method = "none",
                               saveRes = TRUE)
```

**Note:** In the Wilcoxon test, the warning message “**cannot compute
exact p-value with ties.**” may be displayed. This warning means that
some values with tied rankings and the sample size is lower than 50,
which prevents the exact p-value from being calculated. In such cases, a
normal approximation is used. If all samples for a protein have the same
value, the corresponding p-value returns NaN.

    #> $`100pmol-50pmol`

|  | NUD4B_HUMAN (+1) | A0A7P0T808_HUMAN (+1) | A0A8I5KU53_HUMAN (+1) | ZN840_HUMAN | CC85C_HUMAN | C9JEV0_HUMAN (+1) | C9JNU9_HUMAN | ALBU_BOVIN | CYC_BOVIN | TRFE_BOVIN | F8W0H2_HUMAN | H0Y7V7_HUMAN (+1) | H0YD14_HUMAN | H3BUF6_HUMAN | H7C1W4_HUMAN (+1) | H7C3M7_HUMAN | TLR3_HUMAN | LRIG2_HUMAN | RAB3D_HUMAN | ADH1_YEAST | LYSC_CHICK | BGAL_ECOLI | CYTA_HUMAN | KPCB_HUMAN | LIPL_HUMAN | CO6_HUMAN | BGAL_HUMAN | SYTC_HUMAN | CASPE_HUMAN | DCAF6_HUMAN | DALD3_HUMAN | HGNAT_HUMAN | RFFL_HUMAN | RN185_HUMAN | ZN462_HUMAN | ALKB7_HUMAN | POLK_HUMAN | ACAD8_HUMAN |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| 100pmol mean | 10.6532858 | 12.086207 | 10.6232407 | 8.2644365 | 8.8486965 | 6.8764321 | 8.1789684 | 16.75777 | 13.0941802 | 14.0233390 | 9.0659703 | 10.0672927 | 9.9408634 | 7.9781022 | 7.8297313 | 9.8230915 | 7.4096023 | 8.5986327 | 10.0905795 | 15.1604531 | 13.6952175 | 14.3757288 | 8.8924636 | 9.8735187 | 8.2974346 | 6.7681823 | 11.7142153 | 14.7560796 | 6.384893 | 7.9228836 | 9.6214990 | 10.6043210 | 7.0043419 | 10.0418355 | 8.6778910 | 6.8092287 | 7.8188307 | 7.5025180 |
| 50pmol mean | 11.0652499 | 7.812534 | 9.9553033 | 7.9185075 | 8.6105032 | 8.4512816 | 8.3913880 | 16.75777 | 12.6233713 | 13.9262822 | 9.6052951 | 9.6068208 | 10.3568778 | 8.5314736 | 6.9189678 | 9.8860972 | 7.1716461 | 8.6447744 | 10.4687736 | 14.6152277 | 13.4461368 | 14.2433512 | 10.2782600 | 10.2093348 | 8.6777501 | 7.2864883 | 11.7603897 | 15.1244369 | 8.393428 | 7.6429483 | 8.6746639 | 9.8868633 | 6.7178161 | 9.3454998 | 8.2512331 | 7.9571364 | 6.6190600 | 6.8997407 |
| difference | -0.4119641 | 4.273673 | 0.6679373 | 0.3459290 | 0.2381933 | -1.5748494 | -0.2124196 | 0.00000 | 0.4708090 | 0.0970569 | -0.5393248 | 0.4604719 | -0.4160144 | -0.5533715 | 0.9107635 | -0.0630057 | 0.2379563 | -0.0461417 | -0.3781941 | 0.5452255 | 0.2490808 | 0.1323776 | -1.3857964 | -0.3358160 | -0.3803155 | -0.5183059 | -0.0461744 | -0.3683573 | -2.008535 | 0.2799353 | 0.9468351 | 0.7174576 | 0.2865258 | 0.6963357 | 0.4266578 | -1.1479077 | 1.1997707 | 0.6027773 |
| p-value | 0.1912670 | 0.029401 | 0.1885823 | 0.8857143 | 0.1912670 | 0.0571429 | 0.6572552 | NaN | 0.1885823 | 0.1885823 | 0.0284295 | 0.1885823 | 0.0590719 | 0.0294010 | 0.0294010 | 0.6857143 | 0.6631172 | 0.8845494 | 0.0294010 | 0.0255801 | 0.1885823 | 0.1059111 | 0.0285714 | 0.0284295 | 0.1102102 | 0.1102102 | 0.6611967 | 0.0246533 | 0.029401 | 0.3428571 | 0.0590719 | 0.1102102 | 0.3094241 | 0.1142857 | 0.6631172 | 0.0285714 | 0.1102102 | 0.0284295 |

    #> $`200pmol-50pmol`

|  | NUD4B_HUMAN (+1) | A0A7P0T808_HUMAN (+1) | A0A8I5KU53_HUMAN (+1) | ZN840_HUMAN | CC85C_HUMAN | C9JEV0_HUMAN (+1) | C9JNU9_HUMAN | ALBU_BOVIN | CYC_BOVIN | TRFE_BOVIN | F8W0H2_HUMAN | H0Y7V7_HUMAN (+1) | H0YD14_HUMAN | H3BUF6_HUMAN | H7C1W4_HUMAN (+1) | H7C3M7_HUMAN | TLR3_HUMAN | LRIG2_HUMAN | RAB3D_HUMAN | ADH1_YEAST | LYSC_CHICK | BGAL_ECOLI | CYTA_HUMAN | KPCB_HUMAN | LIPL_HUMAN | CO6_HUMAN | BGAL_HUMAN | SYTC_HUMAN | CASPE_HUMAN | DCAF6_HUMAN | DALD3_HUMAN | HGNAT_HUMAN | RFFL_HUMAN | RN185_HUMAN | ZN462_HUMAN | ALKB7_HUMAN | POLK_HUMAN | ACAD8_HUMAN |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| 200pmol mean | 10.6488941 | 9.9995022 | 11.2485095 | 8.4505628 | 8.2445037 | 6.7465842 | 8.3091726 | 16.75777 | 13.2471248 | 13.7426617 | 9.2570565 | 10.1413903 | 9.9978891 | 8.0880583 | 7.5631150 | 10.0415877 | 7.5210813 | 8.4008606 | 10.2414149 | 15.1516556 | 14.1141044 | 14.7415973 | 8.813301 | 10.1732192 | 8.4520910 | 7.0971425 | 12.1775775 | 14.4198014 | 6.1065048 | 7.4951113 | 9.916572 | 10.5958169 | 6.7744045 | 9.4079382 | 8.8671576 | 7.5851679 | 7.2449244 | 8.119644 |
| 50pmol mean | 11.0652499 | 7.8125342 | 9.9553033 | 7.9185075 | 8.6105032 | 8.4512816 | 8.3913880 | 16.75777 | 12.6233713 | 13.9262822 | 9.6052951 | 9.6068208 | 10.3568778 | 8.5314736 | 6.9189678 | 9.8860972 | 7.1716461 | 8.6447744 | 10.4687736 | 14.6152277 | 13.4461368 | 14.2433512 | 10.278260 | 10.2093348 | 8.6777501 | 7.2864883 | 11.7603897 | 15.1244369 | 8.3934284 | 7.6429483 | 8.674664 | 9.8868633 | 6.7178161 | 9.3454998 | 8.2512331 | 7.9571364 | 6.6190600 | 6.899741 |
| difference | -0.4163558 | 2.1869680 | 1.2932062 | 0.5320553 | -0.3659995 | -1.7046973 | -0.0822154 | 0.00000 | 0.6237535 | -0.1836205 | -0.3482386 | 0.5345694 | -0.3589887 | -0.4434153 | 0.6441472 | 0.1554905 | 0.3494352 | -0.2439139 | -0.2273586 | 0.5364280 | 0.6679676 | 0.4982461 | -1.464959 | -0.0361156 | -0.2256591 | -0.1893458 | 0.4171878 | -0.7046356 | -2.2869237 | -0.1478371 | 1.241908 | 0.7089535 | 0.0565885 | 0.0624383 | 0.6159244 | -0.3719685 | 0.6258644 | 1.219903 |
| p-value | 0.2425256 | 0.3094241 | 0.2425256 | 0.1831502 | 0.6572552 | 0.0285714 | 0.8845494 | NaN | 0.1858767 | 0.5589857 | 0.0530079 | 0.1440506 | 0.0575470 | 0.0285714 | 0.3428571 | 0.0814291 | 0.0795941 | 0.3778216 | 0.1831502 | 0.0274686 | 0.0396087 | 0.0265187 | 0.029401 | 1.0000000 | 0.2817179 | 0.4595974 | 0.3035251 | 0.0265187 | 0.0265187 | 0.7715034 | 0.029401 | 0.2000000 | 1.0000000 | 1.0000000 | 0.6631172 | 0.6857143 | 0.2425256 | 0.029401 |

### Details

The Wilcoxon test is a non-parametric alternative to the two-sample
t-test. If `paired = TRUE`, a Wilcoxon signed-rank test is performed to
test the null hypothesis that the distribution of the difference between
the two conditions for the protein is symmetric about zero. If
`paired = FALSE`, a Wilcoxon rank-sum test (also known as Mann-Whitney
test) is performed to test the null hypothesis that the distribution of
the two conditions for the protein are the same.

## MA

### Example

``` r

anlys_ma <- analyze.ma(dataImput, ref = "50pmol", saveRes = TRUE)
```

    #> $`100pmol-50pmol`

|  | NUD4B_HUMAN (+1) | A0A7P0T808_HUMAN (+1) | A0A8I5KU53_HUMAN (+1) | ZN840_HUMAN | CC85C_HUMAN | C9JEV0_HUMAN (+1) | C9JNU9_HUMAN | ALBU_BOVIN | CYC_BOVIN | TRFE_BOVIN | F8W0H2_HUMAN | H0Y7V7_HUMAN (+1) | H0YD14_HUMAN | H3BUF6_HUMAN | H7C1W4_HUMAN (+1) | H7C3M7_HUMAN | TLR3_HUMAN | LRIG2_HUMAN | RAB3D_HUMAN | ADH1_YEAST | LYSC_CHICK | BGAL_ECOLI | CYTA_HUMAN | KPCB_HUMAN | LIPL_HUMAN | CO6_HUMAN | BGAL_HUMAN | SYTC_HUMAN | CASPE_HUMAN | DCAF6_HUMAN | DALD3_HUMAN | HGNAT_HUMAN | RFFL_HUMAN | RN185_HUMAN | ZN462_HUMAN | ALKB7_HUMAN | POLK_HUMAN | ACAD8_HUMAN |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| 100pmol mean | 10.6532858 | 12.086207 | 10.6232407 | 8.264437 | 8.8486965 | 6.876432 | 8.1789684 | 16.75777 | 13.094180 | 14.0233390 | 9.0659703 | 10.0672927 | 9.9408634 | 7.9781022 | 7.8297313 | 9.8230915 | 7.4096023 | 8.5986327 | 10.0905795 | 15.1604531 | 13.6952175 | 14.3757288 | 8.892464 | 9.873519 | 8.2974346 | 6.7681823 | 11.7142153 | 14.7560796 | 6.384893 | 7.9228836 | 9.6214990 | 10.6043210 | 7.0043419 | 10.0418355 | 8.6778910 | 6.809229 | 7.818831 | 7.5025180 |
| 50pmol mean | 11.0652499 | 7.812534 | 9.9553033 | 7.918507 | 8.6105032 | 8.451282 | 8.3913880 | 16.75777 | 12.623371 | 13.9262822 | 9.6052951 | 9.6068208 | 10.3568778 | 8.5314736 | 6.9189678 | 9.8860972 | 7.1716461 | 8.6447744 | 10.4687736 | 14.6152277 | 13.4461368 | 14.2433512 | 10.278260 | 10.209335 | 8.6777501 | 7.2864883 | 11.7603897 | 15.1244369 | 8.393428 | 7.6429483 | 8.6746639 | 9.8868633 | 6.7178161 | 9.3454998 | 8.2512331 | 7.957136 | 6.619060 | 6.8997407 |
| A | 10.8592679 | 9.949371 | 10.2892720 | 8.091472 | 8.7295998 | 7.663857 | 8.2851782 | 16.75777 | 12.858776 | 13.9748106 | 9.3356327 | 9.8370568 | 10.1488706 | 8.2547879 | 7.3743495 | 9.8545943 | 7.2906242 | 8.6217036 | 10.2796765 | 14.8878404 | 13.5706772 | 14.3095400 | 9.585362 | 10.041427 | 8.4875924 | 7.0273353 | 11.7373025 | 14.9402583 | 7.389161 | 7.7829160 | 9.1480815 | 10.2455921 | 6.8610790 | 9.6936677 | 8.4645621 | 7.383183 | 7.218945 | 7.2011293 |
| M | -0.4119641 | 4.273673 | 0.6679373 | 0.345929 | 0.2381933 | -1.574849 | -0.2124196 | 0.00000 | 0.470809 | 0.0970569 | -0.5393248 | 0.4604719 | -0.4160144 | -0.5533715 | 0.9107635 | -0.0630057 | 0.2379563 | -0.0461417 | -0.3781941 | 0.5452255 | 0.2490808 | 0.1323776 | -1.385796 | -0.335816 | -0.3803155 | -0.5183059 | -0.0461744 | -0.3683573 | -2.008535 | 0.2799353 | 0.9468351 | 0.7174576 | 0.2865258 | 0.6963357 | 0.4266578 | -1.147908 | 1.199771 | 0.6027773 |

    #> $`200pmol-50pmol`

|  | NUD4B_HUMAN (+1) | A0A7P0T808_HUMAN (+1) | A0A8I5KU53_HUMAN (+1) | ZN840_HUMAN | CC85C_HUMAN | C9JEV0_HUMAN (+1) | C9JNU9_HUMAN | ALBU_BOVIN | CYC_BOVIN | TRFE_BOVIN | F8W0H2_HUMAN | H0Y7V7_HUMAN (+1) | H0YD14_HUMAN | H3BUF6_HUMAN | H7C1W4_HUMAN (+1) | H7C3M7_HUMAN | TLR3_HUMAN | LRIG2_HUMAN | RAB3D_HUMAN | ADH1_YEAST | LYSC_CHICK | BGAL_ECOLI | CYTA_HUMAN | KPCB_HUMAN | LIPL_HUMAN | CO6_HUMAN | BGAL_HUMAN | SYTC_HUMAN | CASPE_HUMAN | DCAF6_HUMAN | DALD3_HUMAN | HGNAT_HUMAN | RFFL_HUMAN | RN185_HUMAN | ZN462_HUMAN | ALKB7_HUMAN | POLK_HUMAN | ACAD8_HUMAN |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| 200pmol mean | 10.6488941 | 9.999502 | 11.248510 | 8.4505628 | 8.2445037 | 6.746584 | 8.3091726 | 16.75777 | 13.2471248 | 13.7426617 | 9.2570565 | 10.1413903 | 9.9978891 | 8.0880583 | 7.5631150 | 10.0415877 | 7.5210813 | 8.4008606 | 10.2414149 | 15.151656 | 14.1141044 | 14.7415973 | 8.813301 | 10.1732192 | 8.4520910 | 7.0971425 | 12.1775775 | 14.4198014 | 6.106505 | 7.4951113 | 9.916572 | 10.5958169 | 6.7744045 | 9.4079382 | 8.8671576 | 7.5851679 | 7.2449244 | 8.119644 |
| 50pmol mean | 11.0652499 | 7.812534 | 9.955303 | 7.9185075 | 8.6105032 | 8.451282 | 8.3913880 | 16.75777 | 12.6233713 | 13.9262822 | 9.6052951 | 9.6068208 | 10.3568778 | 8.5314736 | 6.9189678 | 9.8860972 | 7.1716461 | 8.6447744 | 10.4687736 | 14.615228 | 13.4461368 | 14.2433512 | 10.278260 | 10.2093348 | 8.6777501 | 7.2864883 | 11.7603897 | 15.1244369 | 8.393428 | 7.6429483 | 8.674664 | 9.8868633 | 6.7178161 | 9.3454998 | 8.2512331 | 7.9571364 | 6.6190600 | 6.899741 |
| A | 10.8570720 | 8.906018 | 10.601906 | 8.1845351 | 8.4275034 | 7.598933 | 8.3502803 | 16.75777 | 12.9352480 | 13.8344719 | 9.4311758 | 9.8741055 | 10.1773834 | 8.3097660 | 7.2410414 | 9.9638424 | 7.3463637 | 8.5228175 | 10.3550942 | 14.883442 | 13.7801206 | 14.4924743 | 9.545781 | 10.1912770 | 8.5649206 | 7.1918154 | 11.9689836 | 14.7721191 | 7.249967 | 7.5690298 | 9.295618 | 10.2413401 | 6.7461103 | 9.3767190 | 8.5591954 | 7.7711521 | 6.9319922 | 7.509692 |
| M | -0.4163558 | 2.186968 | 1.293206 | 0.5320553 | -0.3659995 | -1.704697 | -0.0822154 | 0.00000 | 0.6237535 | -0.1836205 | -0.3482386 | 0.5345694 | -0.3589887 | -0.4434153 | 0.6441472 | 0.1554905 | 0.3494352 | -0.2439139 | -0.2273586 | 0.536428 | 0.6679676 | 0.4982461 | -1.464959 | -0.0361156 | -0.2256591 | -0.1893458 | 0.4171878 | -0.7046356 | -2.286924 | -0.1478371 | 1.241908 | 0.7089535 | 0.0565885 | 0.0624383 | 0.6159244 | -0.3719685 | 0.6258644 | 1.219903 |

### Details

The result of `method = "MA"` is to generate the data for an MA plot,
which plots the average fold change between two conditions (y-axis)
against the average abundance of that protein (x-axis). This is helpful
for evaluating whether a fold-change difference is being enhanced by low
overall intensities (e.g. a change from 200 to 400 is the same
fold-change as from 20,000 to 40,000, but the latter is a more robust
measurement and less susceptible to noise interference).

## PCA

### Example

In the case of `dataImput`, one protein, namely “ALBU_BOVIN”, has
constant values, leading to the error message. We choose to remove this
protein in the principal component analysis (PCA).

``` r

names(dataImput)[sapply(dataImput, function(col) length(unique(col)) == 1)]
#> [1] "ALBU_BOVIN"
dataPCA <- dataImput[, colnames(dataImput) != "ALBU_BOVIN"]
anlys_pca <- analyze.pca(dataPCA, center = TRUE, scale = TRUE)
```

|  | PC1 | PC2 | PC3 | PC4 | PC5 | PC6 | PC7 | PC8 | PC9 | PC10 | PC11 | PC12 |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| NUD4B_HUMAN (+1) | 0.1188769 | -0.1772058 | 0.3013469 | 0.1166326 | -0.1171230 | -0.1140154 | 0.0852809 | -0.0052137 | 0.2575418 | -0.0874986 | 0.2653179 | -0.0436462 |
| A0A7P0T808_HUMAN (+1) | -0.1923637 | 0.2673578 | -0.0130253 | 0.0704243 | 0.1390564 | -0.1433468 | -0.0613349 | -0.0192361 | -0.0110135 | -0.1744425 | 0.0072387 | -0.1108218 |
| A0A8I5KU53_HUMAN (+1) | -0.1289231 | -0.2613306 | 0.0670152 | -0.3011696 | -0.0801726 | 0.0031264 | 0.0131632 | -0.0692313 | -0.0814145 | -0.0598182 | -0.2562817 | 0.1237258 |
| ZN840_HUMAN | -0.0946885 | 0.1276385 | -0.1185999 | -0.0799084 | -0.2528080 | -0.0920193 | -0.4545132 | 0.1824215 | -0.0575034 | 0.2849940 | 0.1804638 | 0.1344257 |
| CC85C_HUMAN | 0.0279395 | 0.1453424 | 0.2659663 | -0.0335283 | -0.2857038 | 0.2540437 | 0.3190762 | -0.0065456 | -0.0636165 | 0.0124279 | -0.0867440 | 0.1462641 |
| C9JEV0_HUMAN (+1) | 0.2195074 | 0.0888077 | -0.0259365 | 0.1759781 | 0.0826490 | 0.1627861 | -0.1151784 | 0.2401487 | 0.1698661 | -0.0474144 | 0.0727410 | 0.1179648 |
| C9JNU9_HUMAN | 0.0506167 | -0.1477131 | 0.0540367 | -0.3813773 | 0.0086479 | -0.0297355 | 0.2233856 | 0.1898863 | -0.1766131 | -0.1663948 | 0.3947327 | -0.2042108 |
| CYC_BOVIN | -0.1529865 | -0.2605458 | 0.1716647 | 0.2067954 | -0.0597879 | 0.0664754 | 0.0508447 | -0.0598310 | -0.0361334 | -0.0545473 | 0.1969247 | 0.0158768 |
| TRFE_BOVIN | 0.0166056 | -0.0168069 | 0.2563064 | 0.2120388 | 0.3158137 | 0.0055484 | -0.1816376 | -0.3163176 | -0.1781309 | 0.0867631 | 0.1887143 | 0.1757929 |
| F8W0H2_HUMAN | 0.2324368 | -0.2037737 | -0.0520720 | -0.0262713 | -0.0057226 | 0.0016452 | -0.1483219 | -0.0509623 | -0.1060480 | 0.1500850 | 0.0055537 | -0.2247142 |
| H0Y7V7_HUMAN (+1) | -0.1988603 | 0.0693049 | -0.1140835 | 0.1625452 | -0.2104495 | 0.2552643 | -0.0405976 | 0.1477780 | -0.0798505 | -0.0086801 | 0.0640366 | 0.1895774 |
| H0YD14_HUMAN | 0.1759816 | -0.1299604 | 0.1545766 | 0.1813077 | -0.1487485 | -0.1570639 | -0.2451852 | 0.1617803 | -0.0081617 | -0.1506331 | -0.0924136 | -0.3292634 |
| H3BUF6_HUMAN | 0.2391701 | -0.0562345 | -0.0842242 | -0.0083829 | 0.2109445 | -0.0478691 | -0.0051651 | 0.1629583 | 0.0829339 | 0.0852264 | -0.2389092 | -0.2402723 |
| H7C1W4_HUMAN (+1) | -0.1571658 | 0.1583107 | -0.0869895 | 0.1324719 | 0.0195078 | 0.4070862 | 0.0582437 | -0.1056046 | 0.2157584 | -0.0357477 | -0.0274390 | -0.1216397 |
| H7C3M7_HUMAN | -0.0268758 | -0.1273659 | -0.1694855 | -0.1822225 | 0.2810524 | 0.1921604 | 0.2311994 | 0.2416035 | -0.0435478 | 0.3395591 | 0.3337349 | 0.0926687 |
| TLR3_HUMAN | -0.1357144 | 0.1117065 | -0.2380256 | 0.1189789 | -0.1890502 | -0.1435597 | 0.1431369 | -0.3576237 | -0.0887085 | 0.0761335 | 0.1526797 | -0.2641330 |
| LRIG2_HUMAN | 0.0995903 | 0.1751785 | 0.1182372 | -0.1571006 | -0.1962309 | 0.3532671 | -0.2061901 | 0.0078915 | 0.2643580 | 0.1422136 | 0.0010136 | -0.1224080 |
| RAB3D_HUMAN | 0.2023449 | -0.1505532 | 0.0064984 | 0.0888776 | -0.2543353 | -0.1184410 | 0.1012599 | 0.0188862 | -0.0559570 | 0.3159153 | -0.1320459 | 0.1405790 |
| ADH1_YEAST | -0.2668869 | 0.0081800 | -0.0189212 | 0.1051980 | 0.0259162 | -0.0238403 | 0.0829892 | 0.0157430 | -0.0507490 | -0.0663319 | -0.1758915 | 0.2215150 |
| LYSC_CHICK | -0.1849233 | -0.2564786 | -0.0640937 | 0.1743153 | -0.0146319 | -0.0206389 | 0.0603589 | -0.0655267 | 0.2384278 | -0.0317316 | 0.1313226 | 0.0502501 |
| BGAL_ECOLI | -0.1785988 | -0.1870954 | -0.2022390 | 0.1424334 | -0.0116692 | 0.0681519 | 0.1151700 | 0.1944421 | 0.2250153 | 0.0457059 | 0.1041234 | -0.0940112 |
| CYTA_HUMAN | 0.2559654 | 0.0494146 | 0.0060060 | 0.1543365 | -0.0138220 | -0.0088571 | -0.0246560 | 0.1585144 | -0.0276568 | 0.1505360 | 0.0409124 | 0.0190272 |
| KPCB_HUMAN | 0.1253089 | -0.0808968 | -0.1725579 | 0.0478656 | -0.2166705 | -0.3758153 | 0.1601399 | -0.1157743 | 0.2303199 | 0.2291294 | -0.0922683 | 0.3117143 |
| LIPL_HUMAN | 0.1525946 | -0.1743527 | -0.0549491 | -0.0480936 | 0.2202094 | 0.1896217 | -0.1268498 | -0.2373904 | 0.4209711 | 0.0006371 | 0.0513403 | 0.1098499 |
| CO6_HUMAN | 0.1695935 | -0.1593740 | -0.2012358 | -0.1658137 | -0.0579254 | 0.0942043 | 0.0302497 | -0.3395673 | 0.1237697 | -0.0386718 | -0.0644920 | -0.0066024 |
| BGAL_HUMAN | -0.0549373 | -0.3398065 | 0.1489976 | 0.1531309 | -0.1653884 | 0.1216312 | 0.0174027 | 0.2036603 | 0.0402230 | -0.0405328 | -0.0633152 | 0.0703951 |
| SYTC_HUMAN | 0.2388616 | 0.1058242 | 0.1738849 | 0.0106153 | 0.0994714 | 0.0475180 | -0.0305461 | -0.0496722 | -0.1463871 | -0.1240736 | 0.0653873 | 0.4380428 |
| CASPE_HUMAN | 0.2394326 | 0.0905518 | -0.0019253 | 0.2112729 | -0.0194369 | 0.0094043 | -0.0243169 | 0.0389020 | -0.0373315 | -0.0237263 | 0.2765286 | 0.0499946 |
| DCAF6_HUMAN | -0.0381553 | 0.1591755 | 0.2198207 | -0.1190328 | 0.3275744 | -0.2690164 | 0.1868800 | 0.0709852 | 0.1486062 | 0.1974166 | 0.0071470 | 0.0550033 |
| DALD3_HUMAN | -0.2266936 | -0.0794761 | 0.0331701 | -0.0648178 | 0.0502899 | -0.1128253 | -0.1004736 | 0.3813898 | 0.0760953 | -0.0233357 | -0.1372768 | 0.0979793 |
| HGNAT_HUMAN | -0.1494193 | -0.2325232 | 0.1921014 | -0.0404533 | -0.0723579 | 0.0148836 | -0.2920131 | -0.1432496 | -0.2087894 | 0.0202744 | 0.1469799 | -0.0242630 |
| RFFL_HUMAN | -0.0806236 | -0.1075798 | 0.1876089 | 0.2489770 | 0.2560138 | 0.1844149 | 0.0279090 | 0.0055965 | -0.1577143 | 0.3487386 | -0.3588672 | -0.1752334 |
| RN185_HUMAN | -0.0969748 | 0.0449635 | 0.3444740 | -0.1393251 | -0.1458840 | 0.0858452 | 0.1314834 | -0.1094733 | 0.0092404 | 0.4236179 | 0.0600972 | -0.1661696 |
| ZN462_HUMAN | -0.1012649 | -0.1599062 | 0.0717821 | -0.3698936 | 0.0658706 | 0.0399805 | -0.2972022 | -0.0645871 | 0.1494307 | -0.0475510 | -0.0870126 | 0.1308837 |
| ALKB7_HUMAN | 0.1400932 | -0.2796786 | -0.1450878 | 0.1117422 | 0.1001606 | 0.1285143 | 0.1473570 | 0.0045354 | -0.2482360 | -0.1677388 | -0.0951268 | 0.0522006 |
| POLK_HUMAN | -0.1657505 | 0.0162727 | 0.2882023 | 0.1061381 | 0.0512940 | -0.2214101 | 0.0338060 | 0.0168708 | 0.3273576 | -0.0500830 | 0.0379469 | -0.0510030 |
| ACAD8_HUMAN | -0.1876209 | -0.1271427 | -0.2161112 | 0.1170599 | 0.1377752 | -0.0681053 | -0.1983427 | -0.0601201 | -0.0648194 | 0.2567213 | 0.1180605 | 0.0198723 |

### Details

PCA is a powerful technique used in data analysis to simplify and reduce
the dimensionality of large datasets. It transforms original variables
into uncorrelated components that capture the maximum variance. By
selecting a subset of these components, PCA projects the data points
onto these key directions, enabling visualization and analysis in a
lower-dimensional space. This aids in identifying patterns and
relationships within complex datasets.

For PCA, the arguments `center` and `scale` are used to center the data
to zero mean and scale to unit variance, with default setting at `TRUE`.

**Note:** Data scaling is done to ensure that the scale differences
between different features do not affect the results of PCA. If not
scaled, features with larger scales will dominate the computation of
principal components (PCs).  
**Note:** The most common error message for the PCA is “**Cannot rescale
a constant/zero column to unit variance**.” This clearly occurs when
columns representing proteins contain only zeros or have constant
values. Typically, there are two ways to address this error: one is to
remove these proteins, and the other is to set `scale = FALSE`.

## PLS-DA

### Example

``` r

anlys_plsda <- analyze.plsda(dataImput, method = "kernelpls",
                             center = TRUE, scale = FALSE)
```

|  | Comp 1 | Comp 2 | Comp 3 | Comp 4 | Comp 5 | Comp 6 | Comp 7 | Comp 8 | Comp 9 | Comp 10 | Comp 11 |
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
| NUD4B_HUMAN (+1) | 0.0721647 | -0.0120098 | 0.1953417 | -0.1265834 | -0.2770654 | 0.2088323 | -0.2009317 | -0.0648547 | 0.0819363 | -0.0454658 | 0.0636382 |
| A0A7P0T808_HUMAN (+1) | -0.7186598 | 0.6655256 | -0.3729955 | -0.0614608 | 0.2153595 | 0.0844135 | -0.1686950 | -0.0100014 | -0.1347304 | 0.0754667 | 0.0196661 |
| A0A8I5KU53_HUMAN (+1) | -0.1319886 | -0.5206130 | 0.3662075 | -0.1771716 | 0.1953444 | -0.1257639 | -0.0554701 | 0.0647629 | -0.2340413 | 0.1951236 | -0.0178531 |
| ZN840_HUMAN | -0.0759383 | 0.0051305 | -0.1438278 | -0.1839000 | -0.0285629 | -0.1132148 | 0.7376771 | -0.5636673 | -0.0141927 | 0.0786254 | 0.0553176 |
| CC85C_HUMAN | 0.0193422 | 0.0726843 | 0.2930632 | 0.2616026 | -0.2469929 | -0.1845150 | 0.0807948 | 0.0392711 | -0.2827831 | 0.0400909 | -0.0442734 |
| C9JEV0_HUMAN (+1) | 0.2703730 | 0.2837151 | -0.1926834 | 0.0625934 | -0.0185917 | 0.1284127 | -0.0099575 | -0.2120263 | 0.2746758 | 0.3386334 | -0.5534373 |
| C9JNU9_HUMAN | 0.0210992 | -0.0655873 | 0.0920396 | -0.1411416 | 0.0807395 | -0.0782736 | -0.0859163 | -0.0237348 | -0.1772656 | -0.3822360 | -0.1003000 |
| ALBU_BOVIN | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |
| CYC_BOVIN | -0.0630095 | -0.1487569 | 0.0827505 | 0.2645965 | -0.3231046 | 0.3509137 | -0.2245885 | -0.0907853 | 0.0138095 | -0.1201481 | 0.0574934 |
| TRFE_BOVIN | -0.0003603 | 0.0327004 | 0.0563023 | 0.0921118 | 0.0790271 | 0.1782056 | -0.0577101 | -0.0134763 | 0.1384664 | -0.0601031 | 0.0685727 |
| F8W0H2_HUMAN | 0.0836310 | -0.0409287 | -0.0097821 | -0.0467141 | 0.0900126 | 0.0470618 | 0.0413209 | -0.0304372 | 0.0481555 | 0.0222324 | 0.0601126 |
| H0Y7V7_HUMAN (+1) | -0.0661155 | -0.0130440 | -0.1035646 | 0.2440707 | -0.1735985 | -0.0442730 | 0.1092398 | -0.1508836 | -0.1183611 | 0.0666902 | -0.0792505 |
| H0YD14_HUMAN | 0.0643955 | 0.0066205 | 0.0315022 | -0.1133884 | -0.0765332 | 0.1873019 | 0.0368123 | -0.1086267 | -0.0615209 | 0.2608078 | 0.0311635 |
| H3BUF6_HUMAN | 0.0817169 | 0.0172734 | -0.0456413 | -0.1147691 | 0.1181330 | 0.0299963 | 0.0002947 | 0.1473198 | 0.0783225 | 0.0595555 | -0.0939223 |
| H7C1W4_HUMAN (+1) | -0.1101453 | 0.0652563 | -0.1173529 | 0.5945244 | -0.1593862 | -0.2350004 | -0.1490996 | -0.0677561 | 0.2601399 | 0.2009991 | -0.2223581 |
| H7C3M7_HUMAN | 0.0034187 | -0.0629444 | -0.0597674 | 0.0426410 | 0.0879519 | -0.0625672 | 0.0024389 | 0.1157673 | 0.0775421 | -0.4824154 | -0.2049414 |
| TLR3_HUMAN | -0.0441759 | 0.0098471 | -0.1081594 | 0.0667498 | -0.0814554 | -0.0948429 | -0.0060339 | 0.0128601 | -0.0473093 | -0.0720274 | 0.2667861 |
| LRIG2_HUMAN | 0.0260332 | 0.0370571 | 0.0887307 | 0.0548165 | -0.0055628 | -0.1406393 | 0.1311827 | -0.1666427 | 0.1044586 | 0.1446907 | -0.0941133 |
| RAB3D_HUMAN | 0.0636594 | -0.0205192 | 0.0081626 | -0.0408032 | -0.0886833 | 0.0241637 | 0.1057128 | 0.0619038 | -0.0443293 | 0.0400687 | 0.1096532 |
| ADH1_YEAST | -0.0886732 | -0.0249521 | -0.0403430 | 0.0903633 | -0.0779594 | 0.0441123 | -0.0239604 | 0.0747446 | -0.0543856 | 0.0371257 | -0.0134305 |
| LYSC_CHICK | -0.0559316 | -0.1146442 | -0.0857625 | 0.0670823 | -0.2123522 | 0.1175924 | -0.1913011 | 0.0088550 | 0.1390133 | -0.0373356 | 0.0392854 |
| BGAL_ECOLI | -0.0354376 | -0.0711086 | -0.1189625 | 0.0522730 | -0.1471152 | 0.0179722 | -0.0593368 | 0.0304717 | 0.0460744 | -0.0709726 | -0.0830799 |
| CYTA_HUMAN | 0.2408724 | 0.1808954 | -0.0650776 | -0.0752154 | -0.0405000 | 0.1463552 | 0.2409196 | 0.0368326 | -0.0089404 | 0.0389971 | -0.0296202 |
| KPCB_HUMAN | 0.0378512 | -0.0135789 | -0.0613165 | -0.1643575 | -0.1085606 | -0.0451864 | 0.0301777 | 0.1125958 | 0.0218511 | 0.0296297 | 0.1979495 |
| LIPL_HUMAN | 0.0574665 | -0.0353568 | -0.0050660 | 0.0030499 | 0.1003712 | -0.0147210 | -0.1869732 | -0.0219275 | 0.3410786 | 0.0676615 | -0.0420835 |
| CO6_HUMAN | 0.0732950 | -0.0688572 | -0.0320353 | -0.0213450 | 0.1139149 | -0.1666744 | -0.1351959 | 0.0021596 | 0.1066948 | 0.0883733 | 0.1395078 |
| BGAL_HUMAN | 0.0143422 | -0.2101695 | 0.0838454 | 0.1545052 | -0.3928935 | 0.3207376 | -0.0666900 | -0.0715498 | -0.0912390 | 0.1973253 | -0.1710358 |
| SYTC_HUMAN | 0.0756364 | 0.0885254 | 0.0966614 | -0.0037612 | 0.1212615 | 0.0467116 | -0.0428996 | -0.0263406 | -0.0298009 | 0.0069434 | -0.0031882 |
| CASPE_HUMAN | 0.3483297 | 0.3750765 | -0.1535783 | 0.0477597 | -0.1098289 | 0.2295415 | -0.0611326 | -0.2841158 | -0.0494692 | -0.1608693 | 0.1757337 |
| DCAF6_HUMAN | -0.0418068 | 0.0849396 | 0.1310571 | -0.1964884 | 0.0685438 | 0.0520614 | -0.0021802 | 0.2694899 | 0.1406201 | -0.3056558 | -0.0866475 |
| DALD3_HUMAN | -0.1896420 | -0.1432009 | -0.0340628 | -0.2470246 | -0.1349039 | 0.1876184 | 0.1936865 | 0.0137017 | -0.0889047 | 0.0389107 | -0.4605911 |
| HGNAT_HUMAN | -0.1046402 | -0.2416903 | 0.2325612 | 0.1301691 | 0.0327822 | 0.3781608 | 0.1030650 | -0.4802290 | 0.0558312 | -0.0006821 | 0.1799745 |
| RFFL_HUMAN | -0.0181685 | -0.0152259 | 0.0389704 | 0.2295137 | -0.0023865 | 0.2135595 | 0.0617714 | 0.1891520 | 0.1219770 | 0.0433609 | -0.0823849 |
| RN185_HUMAN | -0.0585569 | -0.0112455 | 0.3784554 | 0.1343171 | -0.1741584 | -0.0543047 | 0.2908758 | 0.0307732 | 0.0956419 | -0.2721382 | 0.0447837 |
| ZN462_HUMAN | -0.0964568 | -0.2647605 | 0.2505166 | -0.3012533 | 0.4054542 | -0.1195386 | 0.0206231 | -0.3445674 | 0.3837608 | 0.2115073 | -0.2285650 |
| ALKB7_HUMAN | 0.1715043 | -0.1780380 | -0.2018088 | 0.2414476 | 0.1568775 | 0.2092464 | -0.4265611 | 0.2582000 | -0.2462521 | -0.0188007 | -0.0232842 |
| POLK_HUMAN | -0.1889136 | 0.0622625 | 0.2948281 | -0.2461720 | -0.4850179 | 0.3996914 | -0.2316628 | 0.1115249 | 0.3279035 | 0.0315702 | -0.0332998 |
| ACAD8_HUMAN | -0.1306602 | -0.1492646 | -0.3520092 | 0.1200877 | 0.0612338 | 0.2134321 | 0.1391003 | -0.0406605 | 0.3262590 | -0.2431010 | 0.1337232 |

### Details

Partial least squares-discriminant analysis (PLS-DA) adapts PLS
regression for supervised classification. Rather than simply finding
directions of maximal variances in the predictors as PCA does, PLS-DA
extracts latent components that maximize the covariance between
predictors and dummy-coded group labels. This ensures that the resulting
components optimally separate predefined groups and yields
variable-importance scores directly tied to classification.

For PLS-DA, the argument `method` specifies which multivariate
regression algorithm to use:

1.  “kernelpls”: Kernel algorithm (Dayal and MacGregor 1997).

2.  “widekernelpls”: Wide kernel algorithm (Rännar et al. 1994).

3.  “simpls”: SIMPLS algorithm (Jong 1993).

4.  “oscorespls”: NIPALS algorithm (classical orthogonal scores
    algorithm) (Martens and Næs 1989).

The argument `ncomp` sets the number of components to include in the
model. It defaults to min(n-1, p). The arguments `center` and `scale`
control whether the data are centered to zero mean and scaled to unit
variance, respectively.

## Reference

[←
Previous](https://uconn-scs.github.io/msDiaLogue/articles/summarization.md)

[Next
→](https://uconn-scs.github.io/msDiaLogue/articles/visualization.md)

Benjamini, Yoav, and Yosef Hochberg. 1995. “Controlling the False
Discovery Rate: A Practical and Powerful Approach to Multiple Testing.”
*Journal of the Royal Statistical Society: Series B (Methodological)* 57
(1): 289–300. <https://doi.org/10.1111/j.2517-6161.1995.tb02031.x>.

Benjamini, Yoav, and Daniel Yekutieli. 2001. “The Control of the False
Discovery Rate in Multiple Testing Under Dependency.” *The Annals of
Statistics* 29 (4): 1165–88. <https://doi.org/10.1214/aos/1013699998>.

Bonferroni, C. E. 1936. “Teoria Statistica Delle Classi e Calcolo Delle
Probabilità.” *Pubblicazioni Del R Istituto Superiore Di Scienze
Economiche e Commerciali Di Firenze* 8: 3–62.

Dayal, Bhupinder S., and John F. MacGregor. 1997. “Improved PLS
Algorithms.” *Journal of Chemometrics* 11 (1): 73–85.
[https://doi.org/10.1002/(SICI)1099-128X(199701)11:1\<73::AID-CEM435\>3.0.CO;2-\\23](https://doi.org/10.1002/(SICI)1099-128X(199701)11:1%3C73::AID-CEM435%3E3.0.CO;2-\%23).

Hochberg, Yosef. 1988. “A Sharper Bonferroni Procedure for Multiple
Tests of Significance.” *Biometrika* 75 (4): 800–802.
<https://doi.org/10.1093/biomet/75.4.800>.

Holm, Sture. 1979. “A Simple Sequentially Rejective Multiple Test
Procedure.” *Scandinavian Journal of Statistics* 6 (2): 65–70.
<https://www.jstor.org/stable/4615733>.

Hommel, Gerhard. 1988. “A Stagewise Rejective Multiple Test Procedure
Based on a Modified Bonferroni Test.” *Biometrika* 75 (2): 383–86.
<https://doi.org/10.1093/biomet/75.2.383>.

Jong, Sijmen de. 1993. “SIMPLS: An Alternative Approach to Partial Least
Squares Regression.” *Chemometrics and Intelligent Laboratory Systems*
18 (3): 251–63. <https://doi.org/10.1016/0169-7439(93)85002-X>.

Martens, Harald, and Tormod Næs. 1989. *Multivariate Calibration*.
Chichester.

Rännar, Stefan, Fredrik Lindgren, Paul Geladi, and Svante Wold. 1994. “A
PLS Kernel Algorithm for Data Sets with Many Variables and Fewer
Objects. Part 1: Theory and Algorithm.” *Journal of Chemometrics* 8 (2):
111–25. <https://doi.org/10.1002/cem.1180080204>.

Smyth, Gordon K. 2004. “Linear Models and Empirical Bayes Methods for
Assessing Differential Expression in Microarray Experiments.”
*Statistical Applications in Genetics and Molecular Biology* 3 (1).
<https://doi.org/10.2202/1544-6115.1027>.
