<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Imputation • msDiaLogue</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Imputation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">msDiaLogue</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.7</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-modules" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-solid fa-hand-spock"></span> Modules</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-modules">
<li><a class="dropdown-item" href="../articles/preprocessing.html">Preprocessing</a></li>
    <li><a class="dropdown-item" href="../articles/transformation.html">Transformation</a></li>
    <li><a class="dropdown-item" href="../articles/filtering.html">Filtering</a></li>
    <li><a class="dropdown-item" href="../articles/normalization.html">Normalization</a></li>
    <li><a class="dropdown-item" href="../articles/imputation.html">Imputation</a></li>
    <li><a class="dropdown-item" href="../articles/summarization.html">Summarization</a></li>
    <li><a class="dropdown-item" href="../articles/analysis.html">Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization</a></li>
    <li><a class="dropdown-item" href="../articles/other.html">Other Useful</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-solid fa-book"></span> Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="../articles/scaffold.html">Scaffold Export for msDiaLogue</a></li>
    <li><a class="dropdown-item" href="../articles/cust_vis.html">Customizing msDiaLogue Visualizations</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../articles/faq.html"><span class="fa fa-solid fa-clipboard-question"></span> FAQ</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/uconn-scs/msDiaLogue/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Imputation</h1>
                        <h4 data-toc-skip class="author"></h4>
<p>Shiying
Xiao<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>,
Charles
Watt<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>,
Jennifer C.
Liddle<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>2</mn></msup><annotation encoding="application/x-tex">^2</annotation></semantics></math>,
Jeremy L.
Balsbaugh<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>2</mn></msup><annotation encoding="application/x-tex">^2</annotation></semantics></math>,
Timothy E.
Moore<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>3</mn></msup><annotation encoding="application/x-tex">^3</annotation></semantics></math></p>
                        <h4 data-toc-skip class="author"></h4>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>Department
of Statistics, UConn<br><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>2</mn></msup><annotation encoding="application/x-tex">^2</annotation></semantics></math>Proteomics
and Metabolomics Facility, UConn<br><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>3</mn></msup><annotation encoding="application/x-tex">^3</annotation></semantics></math>Statistical
Consulting Services, UConn</p>
            
            <h4 data-toc-skip class="date">2026-02-11</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/uconn-scs/msDiaLogue/blob/HEAD/vignettes/imputation.Rmd" class="external-link"><code>vignettes/imputation.Rmd</code></a></small>
      <div class="d-none name"><code>imputation.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="preliminary">Preliminary<a class="anchor" aria-label="anchor" href="#preliminary"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## load R package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/uconn-scs/msDiaLogue" class="external-link">msDiaLogue</a></span><span class="op">)</span></span>
<span><span class="co">## preprocessing</span></span>
<span><span class="va">fileName</span> <span class="op">&lt;-</span> <span class="st">"../inst/extdata/Toy_Spectronaut_Data.csv"</span></span>
<span><span class="va">dataSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocessing.html">preprocessing</a></span><span class="op">(</span><span class="va">fileName</span>,</span>
<span>                         filterNaN <span class="op">=</span> <span class="cn">TRUE</span>, filterUnique <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                         replaceBlank <span class="op">=</span> <span class="cn">TRUE</span>, saveRm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## transformation</span></span>
<span><span class="va">dataTran</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transform.html">transform</a></span><span class="op">(</span><span class="va">dataSet</span>, logFold <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## normalization</span></span>
<span><span class="va">dataNorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">dataTran</span>, normalizeType <span class="op">=</span> <span class="st">"quant"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>For example, to impute the NA value of <code>dataNorm</code> using
<code><a href="../reference/impute.min_local.html">impute.min_local()</a></code>, set the required percentage of values
that must be present in a given protein by condition combination for
values to be imputed to 51%.</p>
<div class="note">
<strong>Note:</strong> There is no rule in the field of proteomics for
filtering based on percentage of missingness, similar to there being no
rule for the number of replicates required to draw a conclusion.
However, reproducible observations make conclusions more credible.
Setting the <code>reqPercentPresent</code> to 0.51 requires that any
protein be observed in a majority of the replicates by condition in
order to be considered. For 3 replicates, this would require 2
measurements to allow imputation of the 3rd value. If only 1 measurement
is seen, the other values will remain NA, and will be filtered out in a
subsequent step.
</div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataImput_pre</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterNA.html">filterNA</a></span><span class="op">(</span><span class="va">dataNorm</span>, minProp <span class="op">=</span> <span class="fl">0.51</span>, by <span class="op">=</span> <span class="st">"cond"</span>, saveRm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>After the above filtering, any entries that did not pass the percent
present threshold will still have NA values and will need to be filtered
out. And <code>saveRm = TRUE</code> indicates that the filtered data
will be saved as a .csv file named <em>filtered_NA_data.csv</em> in the
current working directory.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataImput</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impute.min_local.html">impute.min_local</a></span><span class="op">(</span><span class="va">dataImput_pre</span><span class="op">)</span></span></code></pre></div>
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="2%">
<col width="2%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="2%">
<col width="2%">
<col width="3%">
<col width="2%">
<col width="2%">
<col width="1%">
<col width="2%">
<col width="2%">
<col width="3%">
<col width="2%">
<col width="2%">
<col width="3%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="1%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="2%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.Replicate</th>
<th align="right">NUD4B_HUMAN (+1)</th>
<th align="right">A0A7P0T808_HUMAN (+1)</th>
<th align="right">A0A8I5KU53_HUMAN (+1)</th>
<th align="right">ZN840_HUMAN</th>
<th align="right">CC85C_HUMAN</th>
<th align="right">C9JEV0_HUMAN (+1)</th>
<th align="right">C9JNU9_HUMAN</th>
<th align="right">ALBU_BOVIN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">TRFE_BOVIN</th>
<th align="right">F8W0H2_HUMAN</th>
<th align="right">H0Y7V7_HUMAN (+1)</th>
<th align="right">H0YD14_HUMAN</th>
<th align="right">H3BUF6_HUMAN</th>
<th align="right">H7C1W4_HUMAN (+1)</th>
<th align="right">H7C3M7_HUMAN</th>
<th align="right">TLR3_HUMAN</th>
<th align="right">LRIG2_HUMAN</th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_ECOLI</th>
<th align="right">CYTA_HUMAN</th>
<th align="right">KPCB_HUMAN</th>
<th align="right">LIPL_HUMAN</th>
<th align="right">CO6_HUMAN</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CASPE_HUMAN</th>
<th align="right">DCAF6_HUMAN</th>
<th align="right">DALD3_HUMAN</th>
<th align="right">HGNAT_HUMAN</th>
<th align="right">RFFL_HUMAN</th>
<th align="right">RN185_HUMAN</th>
<th align="right">ZN462_HUMAN</th>
<th align="right">ALKB7_HUMAN</th>
<th align="right">POLK_HUMAN</th>
<th align="right">ACAD8_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100pmol</td>
<td align="left">1</td>
<td align="right">10.37045</td>
<td align="right">11.406514</td>
<td align="right">10.956950</td>
<td align="right">8.392426</td>
<td align="right">8.710518</td>
<td align="right">7.829510</td>
<td align="right">8.023133</td>
<td align="right">16.75777</td>
<td align="right">12.96499</td>
<td align="right">13.97388</td>
<td align="right">9.136271</td>
<td align="right">10.231965</td>
<td align="right">10.048461</td>
<td align="right">8.179306</td>
<td align="right">8.279169</td>
<td align="right">9.874410</td>
<td align="right">7.001503</td>
<td align="right">8.832972</td>
<td align="right">9.978488</td>
<td align="right">15.16303</td>
<td align="right">13.62766</td>
<td align="right">14.44005</td>
<td align="right">8.964155</td>
<td align="right">9.574185</td>
<td align="right">8.517979</td>
<td align="right">6.764393</td>
<td align="right">12.07953</td>
<td align="right">14.76033</td>
<td align="right">6.004586</td>
<td align="right">7.670711</td>
<td align="right">10.129049</td>
<td align="right">10.681337</td>
<td align="right">7.242036</td>
<td align="right">9.727210</td>
<td align="right">9.376507</td>
<td align="right">7.109682</td>
<td align="right">7.393910</td>
<td align="right">7.530379</td>
</tr>
<tr class="even">
<td align="left">100pmol</td>
<td align="left">2</td>
<td align="right">11.40651</td>
<td align="right">12.964987</td>
<td align="right">9.727210</td>
<td align="right">8.517979</td>
<td align="right">8.832972</td>
<td align="right">7.242036</td>
<td align="right">8.023133</td>
<td align="right">16.75777</td>
<td align="right">13.62766</td>
<td align="right">14.20112</td>
<td align="right">8.964155</td>
<td align="right">10.048461</td>
<td align="right">10.231965</td>
<td align="right">7.829510</td>
<td align="right">7.670711</td>
<td align="right">9.574185</td>
<td align="right">7.393910</td>
<td align="right">8.610420</td>
<td align="right">10.129049</td>
<td align="right">15.16303</td>
<td align="right">13.97388</td>
<td align="right">14.44005</td>
<td align="right">9.136271</td>
<td align="right">9.978488</td>
<td align="right">8.279169</td>
<td align="right">6.420716</td>
<td align="right">12.07953</td>
<td align="right">14.76033</td>
<td align="right">7.109682</td>
<td align="right">8.179306</td>
<td align="right">9.874410</td>
<td align="right">10.956950</td>
<td align="right">7.001503</td>
<td align="right">10.370449</td>
<td align="right">8.392426</td>
<td align="right">6.004586</td>
<td align="right">9.376507</td>
<td align="right">7.530379</td>
</tr>
<tr class="odd">
<td align="left">100pmol</td>
<td align="left">3</td>
<td align="right">10.32522</td>
<td align="right">11.893804</td>
<td align="right">10.851852</td>
<td align="right">7.868171</td>
<td align="right">8.887142</td>
<td align="right">6.429596</td>
<td align="right">8.646475</td>
<td align="right">16.75777</td>
<td align="right">12.81909</td>
<td align="right">13.94448</td>
<td align="right">9.027184</td>
<td align="right">9.940284</td>
<td align="right">9.504539</td>
<td align="right">8.074082</td>
<td align="right">7.698334</td>
<td align="right">10.467264</td>
<td align="right">7.413486</td>
<td align="right">8.433160</td>
<td align="right">10.022816</td>
<td align="right">15.15272</td>
<td align="right">13.55168</td>
<td align="right">14.42169</td>
<td align="right">8.758911</td>
<td align="right">9.812352</td>
<td align="right">8.213284</td>
<td align="right">6.777937</td>
<td align="right">11.29130</td>
<td align="right">14.74334</td>
<td align="right">6.004586</td>
<td align="right">8.311138</td>
<td align="right">9.649565</td>
<td align="right">10.097660</td>
<td align="right">7.009435</td>
<td align="right">10.195272</td>
<td align="right">8.550204</td>
<td align="right">7.121143</td>
<td align="right">7.262869</td>
<td align="right">7.555404</td>
</tr>
<tr class="even">
<td align="left">100pmol</td>
<td align="left">4</td>
<td align="right">10.51096</td>
<td align="right">12.079525</td>
<td align="right">10.956950</td>
<td align="right">8.279169</td>
<td align="right">8.964155</td>
<td align="right">6.004586</td>
<td align="right">8.023133</td>
<td align="right">16.75777</td>
<td align="right">12.96499</td>
<td align="right">13.97388</td>
<td align="right">9.136271</td>
<td align="right">10.048461</td>
<td align="right">9.978488</td>
<td align="right">7.829510</td>
<td align="right">7.670711</td>
<td align="right">9.376507</td>
<td align="right">7.829510</td>
<td align="right">8.517979</td>
<td align="right">10.231965</td>
<td align="right">15.16303</td>
<td align="right">13.62766</td>
<td align="right">14.20112</td>
<td align="right">8.710518</td>
<td align="right">10.129049</td>
<td align="right">8.179306</td>
<td align="right">7.109682</td>
<td align="right">11.40651</td>
<td align="right">14.76033</td>
<td align="right">6.420716</td>
<td align="right">7.530379</td>
<td align="right">8.832972</td>
<td align="right">10.681337</td>
<td align="right">6.764393</td>
<td align="right">9.874410</td>
<td align="right">8.392426</td>
<td align="right">7.001503</td>
<td align="right">7.242036</td>
<td align="right">7.393910</td>
</tr>
<tr class="odd">
<td align="left">200pmol</td>
<td align="left">1</td>
<td align="right">10.27762</td>
<td align="right">12.256403</td>
<td align="right">10.413259</td>
<td align="right">8.356482</td>
<td align="right">7.375266</td>
<td align="right">7.098768</td>
<td align="right">8.149770</td>
<td align="right">16.75777</td>
<td align="right">13.10393</td>
<td align="right">14.00189</td>
<td align="right">9.255807</td>
<td align="right">10.077232</td>
<td align="right">9.798890</td>
<td align="right">8.142561</td>
<td align="right">7.974610</td>
<td align="right">10.164570</td>
<td align="right">7.644405</td>
<td align="right">8.253204</td>
<td align="right">9.927121</td>
<td align="right">15.17286</td>
<td align="right">14.22236</td>
<td align="right">14.77650</td>
<td align="right">8.577167</td>
<td align="right">10.010298</td>
<td align="right">8.782531</td>
<td align="right">7.222195</td>
<td align="right">11.51625</td>
<td align="right">14.45755</td>
<td align="right">6.412260</td>
<td align="right">7.506546</td>
<td align="right">9.641587</td>
<td align="right">10.556023</td>
<td align="right">6.751494</td>
<td align="right">8.669939</td>
<td align="right">9.040857</td>
<td align="right">7.792691</td>
<td align="right">6.993948</td>
<td align="right">8.905805</td>
</tr>
<tr class="even">
<td align="left">200pmol</td>
<td align="left">2</td>
<td align="right">10.53171</td>
<td align="right">11.078642</td>
<td align="right">10.729201</td>
<td align="right">8.356482</td>
<td align="right">8.732804</td>
<td align="right">7.026553</td>
<td align="right">8.267212</td>
<td align="right">16.75777</td>
<td align="right">12.49496</td>
<td align="right">13.38772</td>
<td align="right">9.012072</td>
<td align="right">10.120238</td>
<td align="right">9.798890</td>
<td align="right">8.149770</td>
<td align="right">7.964140</td>
<td align="right">9.954832</td>
<td align="right">7.608925</td>
<td align="right">8.620541</td>
<td align="right">10.229206</td>
<td align="right">15.13045</td>
<td align="right">13.88102</td>
<td align="right">14.70670</td>
<td align="right">8.866513</td>
<td align="right">10.377729</td>
<td align="right">8.386314</td>
<td align="right">7.145874</td>
<td align="right">11.59517</td>
<td align="right">14.38206</td>
<td align="right">6.004586</td>
<td align="right">7.766206</td>
<td align="right">9.827232</td>
<td align="right">9.232358</td>
<td align="right">6.448756</td>
<td align="right">9.504539</td>
<td align="right">8.520402</td>
<td align="right">6.807164</td>
<td align="right">7.455730</td>
<td align="right">7.307825</td>
</tr>
<tr class="odd">
<td align="left">200pmol</td>
<td align="left">3</td>
<td align="right">11.05704</td>
<td align="right">8.142561</td>
<td align="right">12.256403</td>
<td align="right">8.356482</td>
<td align="right">8.905805</td>
<td align="right">6.412260</td>
<td align="right">8.669939</td>
<td align="right">16.75777</td>
<td align="right">14.00189</td>
<td align="right">13.70002</td>
<td align="right">9.255807</td>
<td align="right">10.413259</td>
<td align="right">10.164570</td>
<td align="right">7.792691</td>
<td align="right">7.506546</td>
<td align="right">10.010298</td>
<td align="right">7.375266</td>
<td align="right">8.476493</td>
<td align="right">10.277619</td>
<td align="right">15.17286</td>
<td align="right">14.22236</td>
<td align="right">14.77650</td>
<td align="right">8.577167</td>
<td align="right">9.927121</td>
<td align="right">8.253204</td>
<td align="right">6.993948</td>
<td align="right">13.10393</td>
<td align="right">14.45755</td>
<td align="right">6.004586</td>
<td align="right">7.098768</td>
<td align="right">10.077232</td>
<td align="right">11.516245</td>
<td align="right">6.751494</td>
<td align="right">9.798890</td>
<td align="right">9.040857</td>
<td align="right">7.974610</td>
<td align="right">7.222195</td>
<td align="right">7.644405</td>
</tr>
<tr class="even">
<td align="left">200pmol</td>
<td align="left">4</td>
<td align="right">10.72920</td>
<td align="right">8.520402</td>
<td align="right">11.595175</td>
<td align="right">8.732804</td>
<td align="right">7.964140</td>
<td align="right">6.448756</td>
<td align="right">8.149770</td>
<td align="right">16.75777</td>
<td align="right">13.38772</td>
<td align="right">13.88102</td>
<td align="right">9.504539</td>
<td align="right">9.954832</td>
<td align="right">10.229206</td>
<td align="right">8.267212</td>
<td align="right">6.807164</td>
<td align="right">10.036651</td>
<td align="right">7.455730</td>
<td align="right">8.253204</td>
<td align="right">10.531713</td>
<td align="right">15.13045</td>
<td align="right">14.13067</td>
<td align="right">14.70670</td>
<td align="right">9.232358</td>
<td align="right">10.377729</td>
<td align="right">8.386314</td>
<td align="right">7.026553</td>
<td align="right">12.49496</td>
<td align="right">14.38206</td>
<td align="right">6.004586</td>
<td align="right">7.608925</td>
<td align="right">10.120238</td>
<td align="right">11.078642</td>
<td align="right">7.145874</td>
<td align="right">9.658384</td>
<td align="right">8.866513</td>
<td align="right">7.766206</td>
<td align="right">7.307825</td>
<td align="right">8.620541</td>
</tr>
<tr class="odd">
<td align="left">50pmol</td>
<td align="left">1</td>
<td align="right">10.72920</td>
<td align="right">9.232358</td>
<td align="right">7.766206</td>
<td align="right">8.267212</td>
<td align="right">8.732804</td>
<td align="right">9.658384</td>
<td align="right">7.964140</td>
<td align="right">16.75777</td>
<td align="right">12.49496</td>
<td align="right">13.88102</td>
<td align="right">9.504539</td>
<td align="right">10.120238</td>
<td align="right">10.377729</td>
<td align="right">8.520402</td>
<td align="right">7.608925</td>
<td align="right">9.827232</td>
<td align="right">7.455730</td>
<td align="right">8.620541</td>
<td align="right">10.531713</td>
<td align="right">14.70670</td>
<td align="right">13.38772</td>
<td align="right">14.38206</td>
<td align="right">11.078642</td>
<td align="right">10.229206</td>
<td align="right">8.386314</td>
<td align="right">7.026553</td>
<td align="right">11.59517</td>
<td align="right">15.13045</td>
<td align="right">9.954832</td>
<td align="right">7.307825</td>
<td align="right">8.298269</td>
<td align="right">9.012072</td>
<td align="right">6.807164</td>
<td align="right">8.866513</td>
<td align="right">6.448756</td>
<td align="right">8.149770</td>
<td align="right">6.004586</td>
<td align="right">7.145874</td>
</tr>
<tr class="even">
<td align="left">50pmol</td>
<td align="left">2</td>
<td align="right">10.96831</td>
<td align="right">6.004586</td>
<td align="right">10.662903</td>
<td align="right">8.659793</td>
<td align="right">8.785723</td>
<td align="right">8.190682</td>
<td align="right">8.555305</td>
<td align="right">16.75777</td>
<td align="right">12.30540</td>
<td align="right">13.84672</td>
<td align="right">9.753718</td>
<td align="right">9.581714</td>
<td align="right">10.189464</td>
<td align="right">8.429646</td>
<td align="right">7.035806</td>
<td align="right">10.008606</td>
<td align="right">7.159242</td>
<td align="right">9.099265</td>
<td align="right">10.482590</td>
<td align="right">14.36063</td>
<td align="right">13.25790</td>
<td align="right">14.10465</td>
<td align="right">10.086066</td>
<td align="right">10.189464</td>
<td align="right">8.926299</td>
<td align="right">7.637117</td>
<td align="right">11.47571</td>
<td align="right">15.11842</td>
<td align="right">8.017625</td>
<td align="right">7.480137</td>
<td align="right">8.298269</td>
<td align="right">10.329078</td>
<td align="right">6.459113</td>
<td align="right">9.911682</td>
<td align="right">9.362666</td>
<td align="right">7.332126</td>
<td align="right">6.004586</td>
<td align="right">6.822962</td>
</tr>
<tr class="odd">
<td align="left">50pmol</td>
<td align="left">3</td>
<td align="right">11.59517</td>
<td align="right">6.004586</td>
<td align="right">10.729201</td>
<td align="right">6.448756</td>
<td align="right">8.732804</td>
<td align="right">8.149770</td>
<td align="right">8.386314</td>
<td align="right">16.75777</td>
<td align="right">13.38772</td>
<td align="right">14.13067</td>
<td align="right">9.658384</td>
<td align="right">9.362666</td>
<td align="right">10.377729</td>
<td align="right">8.620541</td>
<td align="right">7.026553</td>
<td align="right">9.954832</td>
<td align="right">7.035806</td>
<td align="right">8.429646</td>
<td align="right">10.531713</td>
<td align="right">14.70670</td>
<td align="right">13.88102</td>
<td align="right">14.38206</td>
<td align="right">10.036651</td>
<td align="right">10.229206</td>
<td align="right">9.012072</td>
<td align="right">7.455730</td>
<td align="right">12.49496</td>
<td align="right">15.13045</td>
<td align="right">7.964140</td>
<td align="right">7.766206</td>
<td align="right">8.520402</td>
<td align="right">10.120238</td>
<td align="right">7.145874</td>
<td align="right">9.504539</td>
<td align="right">8.267212</td>
<td align="right">8.866513</td>
<td align="right">7.307825</td>
<td align="right">6.807164</td>
</tr>
<tr class="even">
<td align="left">50pmol</td>
<td align="left">4</td>
<td align="right">10.96831</td>
<td align="right">10.008606</td>
<td align="right">10.662903</td>
<td align="right">8.298269</td>
<td align="right">8.190682</td>
<td align="right">7.806291</td>
<td align="right">8.659793</td>
<td align="right">16.75777</td>
<td align="right">12.30540</td>
<td align="right">13.84672</td>
<td align="right">9.504539</td>
<td align="right">9.362666</td>
<td align="right">10.482590</td>
<td align="right">8.555305</td>
<td align="right">6.004586</td>
<td align="right">9.753718</td>
<td align="right">7.035806</td>
<td align="right">8.429646</td>
<td align="right">10.329078</td>
<td align="right">14.68689</td>
<td align="right">13.25790</td>
<td align="right">14.10465</td>
<td align="right">9.911682</td>
<td align="right">10.189464</td>
<td align="right">8.386314</td>
<td align="right">7.026553</td>
<td align="right">11.47571</td>
<td align="right">15.11842</td>
<td align="right">7.637117</td>
<td align="right">8.017625</td>
<td align="right">9.581714</td>
<td align="right">10.086066</td>
<td align="right">6.459113</td>
<td align="right">9.099265</td>
<td align="right">8.926299</td>
<td align="right">7.480137</td>
<td align="right">7.159242</td>
<td align="right">6.822962</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a>
</h2>
<p>The two primary MS/MS acquisition types implemented in large scale
MS-based proteomics have unique advantages and disadvantages.
Traditional Data-Dependent Acquisition (DDA) methods favor specificity
in MS/MS sampling over comprehensive proteome coverage. Small peptide
isolation windows (&lt;3 m/z) result in MS/MS spectra that contain
fragmentation data from ideally only one peptide. This specificity
promotes clear peptide identifications but comes at the expense of added
scan time. In DDA experiments, the number of peptides that can be
selected for MS/MS is limited by instrument scan speeds and is therefore
prioritized by highest peptide abundance. Low abundance peptides are
sampled less frequently for MS/MS and this can result in variable
peptide coverage and many missing protein data across large sample
datasets.</p>
<p>Data-Independent Acquisition (DIA) methods promote comprehensive
peptide coverage over specificity by sampling many peptides for MS/MS
simultaneously. Sequential and large mass isolation windows (4-50 m/z)
are used to isolate large numbers of peptides at once for concurrent
MS/MS. This produces complicated fragmentation spectra, but these
spectra contain data on every observable peptide. A major disadvantage
with this type of acquisition is that DIA MS/MS spectra are incredibly
complex and difficult to deconvolve. Powerful and relatively new
software programs like Spectronaut are capable of successfully parsing
out which fragment ions came from each co-fragmented peptide using
custom libraries, machine learning algorithms, and precisely determined
retention times or measured ion mobility data. Because all observable
ions are sampled for MS/MS, DIA reduces missingness substantially
compared to DDA, though not entirely.</p>
<p>Various imputation methods have been developed to address the
missing-value issue and assign a reasonable guess of quantitative value
to proteins with missing values. So far, this package provides 10
imputation methods for use:</p>
<ol style="list-style-type: decimal">
<li><p><code><a href="../reference/impute.min_local.html">impute.min_local()</a></code>: Replaces missing values with the
lowest measured value for that protein in that condition.</p></li>
<li><p><code><a href="../reference/impute.min_global.html">impute.min_global()</a></code>: Replaces missing values with
the lowest measured value from any protein found within the entire
dataset.</p></li>
<li><p><code><a href="../reference/impute.knn.html">impute.knn()</a></code>: Replaces missing values using the
k-nearest neighbors algorithm <span class="citation">(Troyanskaya et al.
2001)</span>.</p></li>
<li><p><code><a href="../reference/impute.knn_seq.html">impute.knn_seq()</a></code>: Replaces missing values using the
sequential k-nearest neighbors algorithm <span class="citation">(Kim et
al. 2004)</span>.</p></li>
<li><p><code><a href="../reference/impute.knn_trunc.html">impute.knn_trunc()</a></code>: Replaces missing values using
the truncated k-nearest neighbors algorithm <span class="citation">(Shah
et al. 2017)</span>.</p></li>
<li><p><code><a href="../reference/impute.nuc_norm.html">impute.nuc_norm()</a></code>: Replaces missing values using the
nuclear-norm regularization <span class="citation">(Hastie et al.
2015)</span>.</p></li>
<li><p><code><a href="../reference/impute.mice_cart.html">impute.mice_cart()</a></code>: Replaces missing values using
the classification and regression trees <span class="citation">(Breiman
et al. 1984; <span class="nocase">Doove et al.</span> 2014; <span class="nocase">van Buuren</span> 2018)</span>.</p></li>
<li><p><code><a href="../reference/impute.mice_norm.html">impute.mice_norm()</a></code>: Replaces missing values using
the Bayesian linear regression <span class="citation">(Rubin 1987;
Schafer 1997; <span class="nocase">van Buuren and
Groothuis-Oudshoorn</span> 2011)</span>.</p></li>
<li><p><code><a href="../reference/impute.pca_bayes.html">impute.pca_bayes()</a></code>: Replaces missing values using
the Bayesian principal components analysis <span class="citation">(Oba
et al. 2003)</span>.</p></li>
<li><p><code><a href="../reference/impute.pca_prob.html">impute.pca_prob()</a></code>: Replaces missing values using the
probabilistic principal components analysis <span class="citation">(Stacklies et al. 2007)</span>.</p></li>
</ol>
<p>Additional methods will be added later.</p>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<div class="nav-button left-button">
<p><a href="normalization.html">← Previous</a></p>
</div>
<div class="nav-button right-button">
<p><a href="summarization.html">Next →</a></p>
</div>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-breiman1984classification" class="csl-entry">
Breiman, L., J. Friedman, R. A. Olshen, and C. J. Stone. 1984.
<em>Classification and Regression Trees</em>. Routledge.
</div>
<div id="ref-doove2014recursive" class="csl-entry">
<span class="nocase">Doove, Lisa L., Stef van Buuren, and Elise
Dusseldorp</span>. 2014. <span>“Recursive Partitioning for Missing Data
Imputation in the Presence of Interaction Effects.”</span>
<em>Computational Statistics &amp; Data Analysis</em> 72: 92–104. <a href="https://doi.org/10.1016/j.csda.2013.10.025" class="external-link">https://doi.org/10.1016/j.csda.2013.10.025</a>.
</div>
<div id="ref-hastie2015matrix" class="csl-entry">
Hastie, Trevor, Rahul Mazumder, Jason D. Lee, and Reza Zadeh. 2015.
<span>“Matrix Completion and Low-Rank <span>SVD</span> via Fast
Alternating Least Squares.”</span> <em>Journal of Machine Learning
Research</em> 16 (104): 3367—3402. <a href="http://jmlr.org/papers/v16/hastie15a.html" class="external-link">http://jmlr.org/papers/v16/hastie15a.html</a>.
</div>
<div id="ref-kim2004reuse" class="csl-entry">
Kim, Ki-Yeol, Byoung-Jin Kim, and Gwan-Su Yi. 2004. <span>“Reuse of
Imputed Data in Microarray Analysis Increases Imputation
Efficiency.”</span> <em>BMC Bioinformatics</em> 5: 160. <a href="https://doi.org/10.1186/1471-2105-5-160" class="external-link">https://doi.org/10.1186/1471-2105-5-160</a>.
</div>
<div id="ref-oba2003bayesian" class="csl-entry">
Oba, Shigeyuki, Masa-aki Sato, Ichiro Takemasa, Morito Monden, Ken-ichi
Matsubara, and Shin Ishii. 2003. <span>“A <span>B</span>ayesian Missing
Value Estimation Method for Gene Expression Profile Data.”</span>
<em>Bioinformatics</em> 19 (16): 2088–96. <a href="https://doi.org/10.1093/bioinformatics/btg287" class="external-link">https://doi.org/10.1093/bioinformatics/btg287</a>.
</div>
<div id="ref-rubin1987multiple" class="csl-entry">
Rubin, Donald B. 1987. <em>Multiple Imputation for Nonresponse in
Surveys</em>. John Wiley &amp; Sons.
</div>
<div id="ref-schafer1997analysis" class="csl-entry">
Schafer, Joseph L. 1997. <em>Analysis of Incomplete Multivariate
Data</em>. Chapman &amp; Hall/CRC.
</div>
<div id="ref-shah2017distribution" class="csl-entry">
Shah, Jasmit S., Shesh N. Rai, Andrew P. DeFilippis, Bradford G. Hill,
Aruni Bhatnagar, and Guy N. Brock. 2017. <span>“Distribution Based
Nearest Neighbor Imputation for Truncated High Dimensional Data with
Applications to Pre-Clinical and Clinical Metabolomics Studies.”</span>
<em>BMC Bioinformatics</em> 18: 114. <a href="https://doi.org/10.1186/s12859-017-1547-6" class="external-link">https://doi.org/10.1186/s12859-017-1547-6</a>.
</div>
<div id="ref-stacklies2007pcamethods" class="csl-entry">
Stacklies, Wolfram, Henning Redestig, Matthias Scholz, Dirk Walther, and
Joachim Selbig. 2007. <span>“<span class="nocase">pcaMethods</span>–a
Bioconductor Package Providing <span>PCA</span> Methods for Incomplete
Data.”</span> <em>Bioinformatics</em> 23 (9): 1164–67. <a href="https://doi.org/10.1093/bioinformatics/btm069" class="external-link">https://doi.org/10.1093/bioinformatics/btm069</a>.
</div>
<div id="ref-troyanskaya2001missing" class="csl-entry">
Troyanskaya, Olga, Michael Cantor, Gavin Sherlock, et al. 2001.
<span>“Missing Value Estimation Methods for <span>DNA</span>
Microarrays.”</span> <em>Bioinformatics</em> 17 (6): 520–25. <a href="https://doi.org/10.1093/bioinformatics/17.6.520" class="external-link">https://doi.org/10.1093/bioinformatics/17.6.520</a>.
</div>
<div id="ref-van2018flexible" class="csl-entry">
<span class="nocase">van Buuren, Stef</span>. 2018. <em>Flexible
Imputation of Missing Data</em>. Chapman &amp; Hall/CRC.
</div>
<div id="ref-van2011mice" class="csl-entry">
<span class="nocase">van Buuren, Stef, and Karin
Groothuis-Oudshoorn</span>. 2011. <span>“Mice: <span>M</span>ultivariate
Imputation by Chained Equations in <span>R</span>.”</span> <em>Journal
of Statistical Software</em> 45 (3): 1–67. <a href="https://doi.org/10.18637/jss.v045.i03" class="external-link">https://doi.org/10.18637/jss.v045.i03</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="http://shiying-xiao.com" class="external-link">Shiying Xiao</a>, <a href="https://statsconsulting.uconn.edu/the-consulting-team" class="external-link">Timothy Moore</a>, Charles Watt.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
