<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Visualization • msDiaLogue</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Visualization">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">msDiaLogue</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.7</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-modules" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-solid fa-hand-spock"></span> Modules</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-modules">
<li><a class="dropdown-item" href="../articles/preprocessing.html">Preprocessing</a></li>
    <li><a class="dropdown-item" href="../articles/transformation.html">Transformation</a></li>
    <li><a class="dropdown-item" href="../articles/filtering.html">Filtering</a></li>
    <li><a class="dropdown-item" href="../articles/normalization.html">Normalization</a></li>
    <li><a class="dropdown-item" href="../articles/imputation.html">Imputation</a></li>
    <li><a class="dropdown-item" href="../articles/summarization.html">Summarization</a></li>
    <li><a class="dropdown-item" href="../articles/analysis.html">Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization</a></li>
    <li><a class="dropdown-item" href="../articles/other.html">Other Useful</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true"><span class="fa fa-solid fa-book"></span> Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials">
<li><a class="dropdown-item" href="../articles/scaffold.html">Scaffold Export for msDiaLogue</a></li>
    <li><a class="dropdown-item" href="../articles/cust_vis.html">Customizing msDiaLogue Visualizations</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../articles/faq.html"><span class="fa fa-solid fa-clipboard-question"></span> FAQ</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/uconn-scs/msDiaLogue/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Visualization</h1>
                        <h4 data-toc-skip class="author"></h4>
<p>Shiying
Xiao<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>,
Charles
Watt<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>,
Jennifer C.
Liddle<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>2</mn></msup><annotation encoding="application/x-tex">^2</annotation></semantics></math>,
Jeremy L.
Balsbaugh<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>2</mn></msup><annotation encoding="application/x-tex">^2</annotation></semantics></math>,
Timothy E.
Moore<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>3</mn></msup><annotation encoding="application/x-tex">^3</annotation></semantics></math></p>
                        <h4 data-toc-skip class="author"></h4>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>1</mn></msup><annotation encoding="application/x-tex">^1</annotation></semantics></math>Department
of Statistics, UConn<br><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>2</mn></msup><annotation encoding="application/x-tex">^2</annotation></semantics></math>Proteomics
and Metabolomics Facility, UConn<br><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi></mi><mn>3</mn></msup><annotation encoding="application/x-tex">^3</annotation></semantics></math>Statistical
Consulting Services, UConn</p>
            
            <h4 data-toc-skip class="date">2026-02-11</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/uconn-scs/msDiaLogue/blob/HEAD/vignettes/visualization.Rmd" class="external-link"><code>vignettes/visualization.Rmd</code></a></small>
      <div class="d-none name"><code>visualization.Rmd</code></div>
    </div>

    
    
<p>This module provides a variety of options for getting a global view
of your data, making comparisons, and highlighting trends.</p>
<div class="note">
<strong>Note:</strong> Data visualization is most effective when
illustrating a point or answering a question you have about your data,
not as a means to find a point/question.
</div>
<p>There is a wide variety of plotting options in
<strong>msDiaLogue</strong>. What plots are possible depend on what data
you have and what analyses have been run. See below for more details of
individual plot types.</p>
<div class="section level2">
<h2 id="preliminary">Preliminary<a class="anchor" aria-label="anchor" href="#preliminary"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## load R package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/uconn-scs/msDiaLogue" class="external-link">msDiaLogue</a></span><span class="op">)</span></span>
<span><span class="co">## preprocessing</span></span>
<span><span class="va">fileName</span> <span class="op">&lt;-</span> <span class="st">"../inst/extdata/Toy_Spectronaut_Data.csv"</span></span>
<span><span class="va">dataSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocessing.html">preprocessing</a></span><span class="op">(</span><span class="va">fileName</span>,</span>
<span>                         filterNaN <span class="op">=</span> <span class="cn">TRUE</span>, filterUnique <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                         replaceBlank <span class="op">=</span> <span class="cn">TRUE</span>, saveRm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## transformation</span></span>
<span><span class="va">dataTran</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transform.html">transform</a></span><span class="op">(</span><span class="va">dataSet</span>, logFold <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## normalization</span></span>
<span><span class="va">dataNorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">dataTran</span>, normalizeType <span class="op">=</span> <span class="st">"quant"</span><span class="op">)</span></span>
<span><span class="co">## filtering</span></span>
<span><span class="va">dataImput_pre</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterNA.html">filterNA</a></span><span class="op">(</span><span class="va">dataNorm</span>, minProp <span class="op">=</span> <span class="fl">0.51</span>, by <span class="op">=</span> <span class="st">"cond"</span>, saveRm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## imputation</span></span>
<span><span class="va">dataImput</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impute.min_local.html">impute.min_local</a></span><span class="op">(</span><span class="va">dataImput_pre</span><span class="op">)</span></span>
<span><span class="co">## analysis: MA</span></span>
<span><span class="va">anlys_ma</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze.ma.html">analyze.ma</a></span><span class="op">(</span><span class="va">dataImput</span>, ref <span class="op">=</span> <span class="st">"50pmol"</span><span class="op">)</span></span>
<span><span class="co">## analysis: Empirical Bayes moderated t-test</span></span>
<span><span class="va">anlys_modt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze.mod_t.html">analyze.mod_t</a></span><span class="op">(</span><span class="va">dataImput</span>, ref <span class="op">=</span> <span class="st">"50pmol"</span>, adjust.method <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="co">## analysis: PCA</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dataImput</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">dataImput</span>, <span class="kw">function</span><span class="op">(</span><span class="va">col</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">col</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">dataPCA</span> <span class="op">&lt;-</span> <span class="va">dataImput</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dataImput</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"ALBU_BOVIN"</span><span class="op">]</span></span>
<span><span class="va">anlys_pca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze.pca.html">analyze.pca</a></span><span class="op">(</span><span class="va">dataPCA</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## analysis: PLS-DA</span></span>
<span><span class="va">anlys_plsda</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze.plsda.html">analyze.plsda</a></span><span class="op">(</span><span class="va">dataImput</span>, method <span class="op">=</span> <span class="st">"kernelpls"</span>,</span>
<span>                             center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="boxplot">Boxplot<a class="anchor" aria-label="anchor" href="#boxplot"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.boxplot.html">visualize.boxplot</a></span><span class="op">(</span><span class="va">dataNorm</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 55 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-3-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="distributions-plot">Distributions plot<a class="anchor" aria-label="anchor" href="#distributions-plot"></a>
</h2>
<p>To examine whether missing values are biased towards lower-abundance
proteins, the distribution plot helps clearly distinguish proteins with
and without missing values.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.dist.html">visualize.dist</a></span><span class="op">(</span><span class="va">dataTran</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-4-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>In the left panel, the Missing group is concentrated in the
low-abundance range, showing a sharper peak than the Valid group, whose
distribution is flatter and more diffuse. This indicates that proteins
with missing values generally have lower abundance—a typical pattern in
proteomics and metabolomics, where low-abundance signals are more prone
to missingness (MNAR).</p>
<p>In the right panel, the ECDF reveals the same pattern: the Missing
group rises more quickly and reaches high cumulative probability at
lower abundance levels, confirming that its average-abundance
distribution is left-shifted relative to the Valid group.</p>
<p>If the input is a list of data frames (e.g., before and after
imputation), density plots are arranged side by side to enable direct
comparison of distributional changes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.dist.html">visualize.dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Pre-Imputation"</span> <span class="op">=</span> <span class="va">dataNorm</span>, <span class="st">"Post-Imputation"</span> <span class="op">=</span> <span class="va">dataImput</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-5-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h2>
<p>The package offers two options for plotting the heatmap, regardless
of whether missing values exist.</p>
<ul>
<li>Option 1 utilizes the source package <code>pheatmap</code>, capable
of plotting the dendrogram simultaneously. It is the default choice for
heatmaps in this package.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.heatmap.html">visualize.heatmap</a></span><span class="op">(</span><span class="va">dataImput</span>, pkg <span class="op">=</span> <span class="st">"pheatmap"</span>,</span>
<span>                  cluster_cols <span class="op">=</span> <span class="cn">TRUE</span>, cluster_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  show_colnames <span class="op">=</span> <span class="cn">TRUE</span>, show_rownames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Setting row names on a tibble is deprecated.</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-6-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>When protein names are excessively long, it is recommended to set
<code>show_rownames = FALSE</code> to view the full heatmap.</p>
<p>If the input data contains an NA value, the heatmap will be binary
for missing data patterns, i.e., whether values are present or
absent.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.heatmap.html">visualize.heatmap</a></span><span class="op">(</span><span class="va">dataSet</span>, pkg <span class="op">=</span> <span class="st">"pheatmap"</span>,</span>
<span>                  cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>, cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                  show_colnames <span class="op">=</span> <span class="cn">TRUE</span>, show_rownames <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  show_pct_cols <span class="op">=</span> <span class="cn">FALSE</span>, show_pct_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  show_pct_legend <span class="op">=</span> <span class="cn">TRUE</span>, saveRes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-7-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<ul>
<li>Option 2 use the source package <code>ggplot2</code> to generate a
ggplot object but does not include the dendrogram.</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.heatmap.html">visualize.heatmap</a></span><span class="op">(</span><span class="va">dataImput</span>, pkg <span class="op">=</span> <span class="st">"ggplot2"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Setting row names on a tibble is deprecated.</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-8-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>In a heatmap, similar colors within a row indicate relatively
consistent values, suggesting similar protein expression levels across
different samples.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.heatmap.html">visualize.heatmap</a></span><span class="op">(</span><span class="va">dataSet</span>, pkg <span class="op">=</span> <span class="st">"ggplot2"</span>,</span>
<span>                  show_pct_cols <span class="op">=</span> <span class="cn">FALSE</span>, show_pct_rows <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  show_pct_legend <span class="op">=</span> <span class="cn">TRUE</span>, saveRes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-9-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="ma-plot">MA plot<a class="anchor" aria-label="anchor" href="#ma-plot"></a>
</h2>
<div class="section level3">
<h3 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.ma.html">visualize.ma</a></span><span class="op">(</span><span class="va">anlys_ma</span><span class="op">$</span><span class="va">`100pmol-50pmol`</span>, M.thres <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 32 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text_repel()`).</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-10-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>where <code>M.thres = 1</code> means the M thresholds are set to -1
and 1. The scatters are split into three parts: up regulation (M &gt;
1), no regulation (-1
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>≤</mo><annotation encoding="application/x-tex">\leq</annotation></semantics></math>
M
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>≤</mo><annotation encoding="application/x-tex">\leq</annotation></semantics></math>
1), and down regulation (M &lt; -1). Additionally, the warning message
“Removed 32 rows containing missing values” indicates that there are 32
proteins with no regulation.</p>
<p>If the input <code>dataSet</code> is the whole list
<code>anlys_MA</code>, <strong>msDiaLogue</strong> will produce
individual subplots corresponding to each comparison.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.ma.html">visualize.ma</a></span><span class="op">(</span><span class="va">anlys_ma</span>, M.thres <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 63 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text_repel()`).</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-11-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a>
</h3>
<p>An MA plot visualizes the differences between measurements taken in
two samples, by transforming the data onto M (log ratio or fold change)
and A (mean average) scales. The MA plot puts the variable M on the
y-axis and A on the x-axis and gives a quick overview of the
distribution of the data. Most proteins are expected to show little
variation, and so the majority of the points will be concentrated around
the M = 0 line (no difference between group means). Typically, points
falling above
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>a</mi><mi>b</mi><mi>s</mi><mo stretchy="false" form="prefix">(</mo><mn>1</mn><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">abs(1)</annotation></semantics></math>
are highlighted.</p>
<div class="note">
<strong>Note:</strong> As the MA plot does not include any measure of
statistical significance, it cannot directly identify which proteins are
statistically differentially expressed.
</div>
<p>An MA plot, short for “M vs. A plot,” which uses two axes:</p>
<ul>
<li>M axis (vertical): Represents the fold change, usually on the
logarithm base 2 scale, or the ratio of the expression levels, between
two conditions. It is calculated as:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><mi>l</mi><mi>o</mi><msub><mi>g</mi><mn>2</mn></msub><mo stretchy="false" form="prefix">(</mo><mi>X</mi><mi>/</mi><mi>Y</mi><mo stretchy="false" form="postfix">)</mo><mo>=</mo><mi>l</mi><mi>o</mi><msub><mi>g</mi><mn>2</mn></msub><mi>X</mi><mo>−</mo><mi>l</mi><mi>o</mi><msub><mi>g</mi><mn>2</mn></msub><mi>Y</mi></mrow><annotation encoding="application/x-tex">M = log_2(X/Y) = log_2 X - log_2 Y</annotation></semantics></math>
</li>
<li>A axis (horizontal): Represents the average intensity of the two
conditions, calculated as:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>A</mi><mo>=</mo><mfrac><mn>1</mn><mn>2</mn></mfrac><mi>l</mi><mi>o</mi><msub><mi>g</mi><mn>2</mn></msub><mo stretchy="false" form="prefix">(</mo><mi>X</mi><mi>Y</mi><mo stretchy="false" form="postfix">)</mo><mo>=</mo><mfrac><mn>1</mn><mn>2</mn></mfrac><mrow><mo stretchy="true" form="prefix">[</mo><mi>l</mi><mi>o</mi><msub><mi>g</mi><mn>2</mn></msub><mo stretchy="false" form="prefix">(</mo><mi>X</mi><mo stretchy="false" form="postfix">)</mo><mo>+</mo><mi>l</mi><mi>o</mi><msub><mi>g</mi><mn>2</mn></msub><mo stretchy="false" form="prefix">(</mo><mi>Y</mi><mo stretchy="false" form="postfix">)</mo><mo stretchy="true" form="postfix">]</mo></mrow></mrow><annotation encoding="application/x-tex">A = \frac{1}{2}log_2(XY) = \frac{1}{2}\left[log_2(X)+log_2(Y)\right]</annotation></semantics></math>
</li>
</ul>
<p>Most proteins are expected to exhibit little variation, leading to
the majority of points concentrating around the line M = 0 (indicating
no difference between group means).</p>
</div>
</div>
<div class="section level2">
<h2 id="rank-abundance-distribution-plot-whittaker-plot">Rank abundance distribution plot (Whittaker plot)<a class="anchor" aria-label="anchor" href="#rank-abundance-distribution-plot-whittaker-plot"></a>
</h2>
<p>The rank abundance plot shows proteins ranked by abundance to
visualize distribution patterns and highlight specific proteins across
conditions or replicates.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.rank.html">visualize.rank</a></span><span class="op">(</span><span class="va">dataImput</span>, listName <span class="op">=</span> <span class="st">"POLK_HUMAN"</span>,</span>
<span>               facet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Replicate"</span>, <span class="st">"Condition"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-12-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="histogram-of-fold-changes-and-p-values-for-test">Histogram of fold changes and p-values for test<a class="anchor" aria-label="anchor" href="#histogram-of-fold-changes-and-p-values-for-test"></a>
</h2>
<div class="section level3">
<h3 id="examples-1">Examples<a class="anchor" aria-label="anchor" href="#examples-1"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.test.html">visualize.test</a></span><span class="op">(</span><span class="va">anlys_modt</span><span class="op">$</span><span class="va">`100pmol-50pmol`</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-13-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>If the input <code>dataSet</code> is the whole list
<code>anlys_modt</code>, <strong>msDiaLogue</strong> will produce
individual subplots corresponding to each comparison.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.test.html">visualize.test</a></span><span class="op">(</span><span class="va">anlys_modt</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-14-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="details-1">Details<a class="anchor" aria-label="anchor" href="#details-1"></a>
</h3>
<p>The function <code>visualize().test</code> can be applied to any test
output. It generates two useful plots: a histogram of fold changes
across the analyzed proteins and a histogram of p-values. The majority
of proteins are expected to show very small change between conditions,
so the fold change histogram will have a peak at around zero. For the
p-values, most p-values are expected to be non-significant (above 0.05).
Depending on the strength of the treatment effect, there may be a peak
of p-values near 0.</p>
</div>
</div>
<div class="section level2">
<h2 id="upset-plot">UpSet plot<a class="anchor" aria-label="anchor" href="#upset-plot"></a>
</h2>
<p>The UpSet plot is a visual representation that helps display the
overlap and intersection of sets or categories in a dataset. It is
particularly useful for illustrating the presence or absence of elements
in combinations of sets.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.upset.html">visualize.upset</a></span><span class="op">(</span><span class="va">dataSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-15-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>This plot reveals that 42 proteins are shared by 50pmol, 100pmol, and
200pmol, while only 3 proteins are shared by 100 pmol and 200pmol, but
not with 50pmol.</p>
</div>
<div class="section level2">
<h2 id="venn-plot">Venn plot<a class="anchor" aria-label="anchor" href="#venn-plot"></a>
</h2>
<p>The Venn plot is another graphical representation of the
relationships between sets. Each circle represents a set, and the
overlapping regions show the elements that are shared between sets.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.venn.html">visualize.venn</a></span><span class="op">(</span><span class="va">dataSet</span>, show_percentage <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               fill_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"yellow"</span>, <span class="st">"green"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>               saveRes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-16-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>where <code>saveRes = TRUE</code> refers to the data containing
logical columns representing sets in Venn plot information will be saved
as a .csv file named <em>venn_information.csv</em> in the current
working directory.</p>
<p>In the example above, 50pmol, 100pmol, and 200pmol groups share 42
proteins. Notably, 3 proteins are exclusively found in the 100pmol and
200pmol groups.</p>
</div>
<div class="section level2">
<h2 id="volcano-plot">Volcano plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h2>
<div class="section level3">
<h3 id="examples-2">Examples<a class="anchor" aria-label="anchor" href="#examples-2"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.volcano.html">visualize.volcano</a></span><span class="op">(</span><span class="va">anlys_modt</span><span class="op">$</span><span class="va">`100pmol-50pmol`</span>, P.thres <span class="op">=</span> <span class="fl">0.05</span>, F.thres <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 32 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text_repel()`).</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-17-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>If the input <code>dataSet</code> is the whole list
<code>anlys_modt</code>, <strong>msDiaLogue</strong> will produce
individual subplots corresponding to each comparison.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.volcano.html">visualize.volcano</a></span><span class="op">(</span><span class="va">anlys_modt</span>, P.thres <span class="op">=</span> <span class="fl">0.05</span>, F.thres <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 65 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text_repel()`).</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-18-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="details-2">Details<a class="anchor" aria-label="anchor" href="#details-2"></a>
</h3>
<p>A volcano plot is a graphical representation commonly used in
proteomics and genomics to visualize differential expression analysis
results. It is particularly useful for identifying significant changes
in extensive data. It displays two important pieces of information about
differences between conditions in a dataset:</p>
<ul>
<li><p>Statistical significance (vertical): Represents the negative
log10 of the p-value.</p></li>
<li><p>Fold change (horizontal): Represents the fold change.</p></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="scree-plot">Scree plot<a class="anchor" aria-label="anchor" href="#scree-plot"></a>
</h2>
<p>One way to help identify how many PCs to retain is to explore a scree
plot. The scree plot shows the percentage of variance explained by each
PC.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.scree.html">visualize.scree</a></span><span class="op">(</span><span class="va">anlys_pca</span>, type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bar"</span>, <span class="st">"line"</span><span class="op">)</span>,</span>
<span>                bar.color <span class="op">=</span> <span class="st">"gray"</span>, bar.fill <span class="op">=</span> <span class="st">"gray"</span>, line.color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                label <span class="op">=</span> <span class="cn">TRUE</span>, ncp <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-19-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
<p>where <code>label = TRUE</code> adds information labels at the top of
bars/points, and <code>ncp = 10</code> sets the number of dimension to
be displayed.</p>
</div>
<div class="section level2">
<h2 id="score-plot-graph-of-individuals">Score plot / graph of individuals<a class="anchor" aria-label="anchor" href="#score-plot-graph-of-individuals"></a>
</h2>
<div class="section level3">
<h3 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.score.html">visualize.score</a></span><span class="op">(</span><span class="va">anlys_pca</span>, ellipse <span class="op">=</span> <span class="cn">TRUE</span>, ellipse.level <span class="op">=</span> <span class="fl">0.95</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-20-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="details-3">Details<a class="anchor" aria-label="anchor" href="#details-3"></a>
</h3>
<p>The primary PCA plot of individual data visually represents the
distribution of individual observations in a reduced-dimensional space,
typically defined by the PCs. The x and y axes of the PCA plot represent
the PCs. Each axis corresponds to a linear combination of the original
variables. Individual data points on the PCA plot represent observations
(e.g., samples) from the original dataset. Points that are close to the
origin (0, 0), are close to the “average” across all protein abundances.
If sufficient samples are present, the plot will also produce a 95%
confidence ellipse, as well as a centroid (mean for each group
provided), for each groups (condition) provided.</p>
</div>
</div>
<div class="section level2">
<h2 id="loading-plot-graph-of-variables">Loading plot / graph of variables<a class="anchor" aria-label="anchor" href="#loading-plot-graph-of-variables"></a>
</h2>
<div class="section level3">
<h3 id="example-1">Example<a class="anchor" aria-label="anchor" href="#example-1"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.loading.html">visualize.loading</a></span><span class="op">(</span><span class="va">anlys_pca</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-21-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="details-4">Details<a class="anchor" aria-label="anchor" href="#details-4"></a>
</h3>
<p>This plot will be more useful if your analyses are based on a
relatively small number of proteins. It represents the association, or
loading of each protein on the first two PCs. Longer arrows represents
stronger associations.</p>
<div class="note">
<strong>Note:</strong> Proteins that are weakly associated with PC1 or
PC2 may still be highly correlated with other PCs not being plotted.
Consult the scree plot (and other available methods) to determine the
appropriate number of PCs to investigate.
</div>
</div>
</div>
<div class="section level2">
<h2 id="biplot-of-score-individuals-and-loading-variables">Biplot of score (individuals) and loading (variables)<a class="anchor" aria-label="anchor" href="#biplot-of-score-individuals-and-loading-variables"></a>
</h2>
<p>The biplot includes individual and variable plots. Again, with a
large number of proteins, this plot can be unwieldy.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.biplot.html">visualize.biplot</a></span><span class="op">(</span><span class="va">anlys_pca</span>, ellipse <span class="op">=</span> <span class="cn">TRUE</span>, ellipse.level <span class="op">=</span> <span class="fl">0.95</span>, label <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-22-1.png" class="r-plt" alt="" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="vip-score-plot">VIP score plot<a class="anchor" aria-label="anchor" href="#vip-score-plot"></a>
</h2>
<p>This combined plot pairs a VIP-score chart on the left with a heatmap
on the right. The left panel shows each variable as a dot at its VIP
score from PLS-DA. Higher scores indicate greater importance for class
separation. The right panel uses colored tiles to show each top
variable’s average abundance across conditions. So you can instantly see
which variables matter most for class separation and how they behave in
each condition.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.vip.html">visualize.vip</a></span><span class="op">(</span><span class="va">anlys_plsda</span>, comp <span class="op">=</span> <span class="fl">1</span>, num <span class="op">=</span> <span class="fl">10</span>, thres <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="visualization_files/figure-html/unnamed-chunk-23-1.png" class="r-plt" alt="" width="864" style="display: block; margin: auto;"></p>
<div class="nav-button left-button">
<p><a href="analysis.html">← Previous</a></p>
</div>
<div class="nav-button right-button">
<p><a href="other.html">Next →</a></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="http://shiying-xiao.com" class="external-link">Shiying Xiao</a>, <a href="https://statsconsulting.uconn.edu/the-consulting-team" class="external-link">Timothy Moore</a>, Charles Watt.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
